---
title: "Consensus Clustering - feature level"
author: "Sergio Mosquim Junior"
format:
  html:
    toc: true
    code-fold: true
    embed-resources: true
    toc-location: left
    smooth-scroll: true
engine: knitr
---

# Required Packages
```{r}
library(ConsensusClusterPlus)
library(tidyverse)
library(ComplexHeatmap)
source(file = '/home/multiomics/notebooks/filtering.R',chdir = TRUE)
```

# Consensus Clustering Features
```{r}
if(!dir.exists(paths = '/home/multiomics/results/consensus_clustering/features/')) {
  dir.create(path = '/home/multiomics/results/consensus_clustering/features/',recursive = TRUE)
}
```

### Proteome
#### Load Data
```{r}
data <- read_tsv(file = '/home/multiomics/results/batchcorrection/fullproteome/cycloess_rollup_genes_batchcorrected.tsv') %>% 
  drop_na()
design <- read_tsv(file = '/home/multiomics/results/design_files/design_Full_noPool.tsv')
```

#### Preprocessing
```{r}
proteinDataFiltered <- varianceFiltering(data = data,design = design,sampleCol = 'sample_id',annotation = 'Protein',cutoff = 0.75)

proteinDataReduced <- proteinDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix %>% 
  t()

colnames(proteinDataReduced) <- make.names(proteinDataFiltered[['Protein.Names']],unique = TRUE)
```

#### Plotting
```{r}
setwd('/home/multiomics/results/consensus_clustering/features/')
results <-  ConsensusClusterPlus(d = proteinDataReduced,
                                 maxK = 10,
                                 clusterAlg = 'km',
                                 seed = 123,
                                 verbose = TRUE,reps = 50,plot = 'pdf',writeTable = TRUE,
                                 title = 'proteome')
```


### Phosphoproteome
#### Load Data
```{r}
data <- read_tsv(file = '/home/multiomics/results/batchcorrection/phosphoproteome/cycloess_peptide_batchcorrected.tsv') %>% 
  drop_na
design <- read_tsv(file = '/home/multiomics/results/design_files/design_phospho_noPool.tsv')
```
#### Preprocessing
```{r}
phosphoDataFiltered <- varianceFiltering(data = data,design = design,sampleCol = 'sample_id',annotation = 'Precursor.Id')

phosphoDataReduced <- phosphoDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix() %>% 
  t()

colnames(phosphoDataReduced) <- make.names(names = phosphoDataFiltered[['Protein.Names']],unique = TRUE)
```

#### Plotting
```{r}
setwd('/home/multiomics/results/consensus_clustering/features/')
results <-  ConsensusClusterPlus(d = phosphoDataReduced,
                                 maxK = 10,
                                 clusterAlg = 'km',
                                 seed = 123,
                                 verbose = TRUE,reps = 50,plot = 'pdf',writeTable = TRUE,
                                 title = 'phosphoproteome')
```

### Transcriptome
#### Load Data
```{r}
data <- read_tsv(file = '/home/multiomics/data/transcriptomics/genematrix_adjusted_cycloess.tsv') %>% 
  drop_na
design <- read_tsv(file = '/home/multiomics/results/design_files/design_RNA_noPool.tsv')
```

#### Preprocessing
```{r}
rnaDataFiltered <- varianceFiltering(data = data,design = design,sampleCol = 'sample_id',annotation = 'Gene.ID')

rnaDataReduced <- rnaDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix() %>% 
  t()

colnames(rnaDataReduced) <- make.names(names = rnaDataFiltered[['Gene.ID']],unique = TRUE)
```

#### Plotting
```{r}
setwd('/home/multiomics/results/consensus_clustering/features/')
results <-  ConsensusClusterPlus(d = rnaDataReduced,
                                 maxK = 10,
                                 clusterAlg = 'km',
                                 seed = 123,
                                 verbose = TRUE,reps = 50,plot = 'pdf',writeTable = TRUE,
                                 title = 'transcriptome')
```
