---
title: "Consensus Clustering"
author: "Sergio Mosquim Junior"
format:
  html:
    toc: true
    code-fold: true
    embed-resources: false
    toc-location: left
    smooth-scroll: true
engine: knitr
execute: 
  output: false
params:
  run: false
---

# Required Packages
```{r}
library(ConsensusClusterPlus)
library(tidyverse)
library(ComplexHeatmap)
source(file = '/home/multiomics/notebooks/filtering.R',chdir = TRUE)
```

# Consensus Clustering
## Ductal vs Lobular
```{r}
if(!dir.exists(paths = '/home/multiomics/results/consensus_clustering/DuctalvsLobular/')) {
  dir.create(path = '/home/multiomics/results/consensus_clustering/DuctalvsLobular/',recursive = TRUE)
}
```

### Proteome
#### Load Data
```{r}
data <- read_tsv(file = '/home/multiomics/results/batchcorrection/fullproteome/cycloess_rollup_genes_batchcorrected.tsv') %>% 
  drop_na()
design <- read_tsv(file = '/home/multiomics/results/design_files/design_Full_noPool.tsv')
```

#### Preprocessing
```{r}
proteinDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.75)

proteinDataFiltered <- proteinDataFiltered %>% 
  dplyr::select(Protein,design$sample_id)

proteinDataReduced <- proteinDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
```

#### Plotting
```{r}
setwd('/home/multiomics/results/consensus_clustering/DuctalvsLobular/')
results <-  ConsensusClusterPlus(d = proteinDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'proteome')
```

### Phosphoproteome
#### Load Data
```{r}
data <- read_tsv(file = '/home/multiomics/results/batchcorrection/phosphoproteome/cycloess_peptide_flanking_batchcorrected.tsv') %>% 
  drop_na
design <- read_tsv(file = '/home/multiomics/results/design_files/design_phospho_noPool.tsv')
```

#### Preprocessing
```{r}
phosphoDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.5)

phosphoDataFiltered <- phosphoDataFiltered %>% 
  dplyr::select(Protein.Group,design$sample_id)

phosphoDataReduced <- phosphoDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix()
```

#### Plotting
```{r}
setwd('/home/multiomics/results/consensus_clustering/DuctalvsLobular/')
results <-  ConsensusClusterPlus(d = phosphoDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'phosphoproteome')
```

### Transcriptome
#### Load Data
```{r}
data <- read_tsv(file = '/home/multiomics/results/normalization/transcriptomics/genematrix_adjusted_cycloess.tsv') %>% 
  drop_na
design <- read_tsv(file = '/home/multiomics/results/design_files/design_RNA_noPool.tsv')
```

#### Preprocessing
```{r}
rnaDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.75)

rnaDataFiltered <- rnaDataFiltered %>% 
  dplyr::select(Gene.ID,design$sample_id)

rnaDataReduced <- rnaDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix()
```

#### Plotting
```{r}
setwd('/home/multiomics/results/consensus_clustering/DuctalvsLobular/')
results <-  ConsensusClusterPlus(d = rnaDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'transcriptome')
```

:::{.content-visible when-meta="params.run"}
## All samples - Group2 Excluded
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
if(!dir.exists(paths = '/home/multiomics/results/consensus_clustering/AllSamples_noG2/')) {
  dir.create(path = '/home/multiomics/results/consensus_clustering/AllSamples_noG2/',recursive = TRUE)
}
```

### Proteome
#### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/batchcorrection/fullproteome/cycloess_rollup_genes_batchcorrected.tsv') %>% 
  drop_na()
design <- read_tsv(file = '/home/multiomics/results/design_files/design_Full_noPool_noG2.tsv')
```

#### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
proteinDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.75)

proteinDataFiltered <- proteinDataFiltered %>% 
  dplyr::select(Protein,design$sample_id)

proteinDataReduced <- proteinDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
```

#### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/AllSamples_noG2/')
results <-  ConsensusClusterPlus(d = proteinDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'proteome')
```

### Phosphoproteome
#### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/batchcorrection/phosphoproteome/cycloess_peptide_flanking_batchcorrected.tsv') %>% 
  drop_na
design <- read_tsv(file = '/home/multiomics/results/design_files/design_phospho_noPool_noG2.tsv')
```

#### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
phosphoDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.5)

phosphoDataFiltered <- phosphoDataFiltered %>% 
  dplyr::select(Protein.Group,design$sample_id)

phosphoDataReduced <- phosphoDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix()
```

#### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/AllSamples_noG2/')
results <-  ConsensusClusterPlus(d = phosphoDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'phosphoproteome')
```

### Transcriptome
#### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/normalization/transcriptomics/genematrix_adjusted_cycloess.tsv') %>% 
  drop_na
design <- read_tsv(file = '/home/multiomics/results/design_files/design_RNA_noPool_noG2.tsv')
```

#### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
rnaDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.75)

rnaDataFiltered <- rnaDataFiltered %>% 
  dplyr::select(Gene.ID,design$sample_id)

rnaDataReduced <- rnaDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix()
```

#### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/AllSamples_noG2/')
results <-  ConsensusClusterPlus(d = rnaDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'transcriptome')
```

## LN+ vs LN-
### Ductal
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
if(!dir.exists(paths = '/home/multiomics/results/consensus_clustering/LNposvsLNneg/Ductal/')) {
  dir.create(path = '/home/multiomics/results/consensus_clustering/LNposvsLNneg/Ductal/',recursive = TRUE)
}
```

#### Proteome
##### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/batchcorrection/fullproteome/cycloess_rollup_genes_batchcorrected.tsv') %>% 
  drop_na()
design <- read_tsv(file = '/home/multiomics/results/design_files/design_full_ductal.tsv')
```

##### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
proteinDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.75)

proteinDataFiltered <- proteinDataFiltered %>% 
  dplyr::select(Protein,design$sample_id)

proteinDataReduced <- proteinDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
```

##### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/LNposvsLNneg/Ductal/')
results <-  ConsensusClusterPlus(d = proteinDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'proteome')
```

#### Phosphoproteome
##### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/batchcorrection/phosphoproteome/cycloess_peptide_flanking_batchcorrected.tsv') %>% 
  drop_na
design <- read_tsv(file = '/home/multiomics/results/design_files/design_phospho_ductal.tsv')
```
##### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
phosphoDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.5)

phosphoDataFiltered <- phosphoDataFiltered %>% 
  dplyr::select(Protein.Group,design$sample_id)

phosphoDataReduced <- phosphoDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix()
```

##### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/LNposvsLNneg/Ductal/')
results <-  ConsensusClusterPlus(d = phosphoDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'phosphoproteome')
```

#### Transcriptome
##### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/normalization/transcriptomics/genematrix_adjusted_cycloess.tsv') %>% 
  drop_na
design <- read_tsv(file = '/home/multiomics/results/design_files/design_RNA_ductal.tsv')
```

##### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
rnaDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.75)

rnaDataFiltered <- rnaDataFiltered %>% 
  dplyr::select(Gene.ID,design$sample_id)

rnaDataReduced <- rnaDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix()
```

##### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/LNposvsLNneg/Ductal/')
results <-  ConsensusClusterPlus(d = rnaDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'transcriptome')
```


### Ductal - Group2 Excluded
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
if(!dir.exists(paths = '/home/multiomics/results/consensus_clustering/LNposvsLNneg/Ductal_noG2/')) {
  dir.create(path = '/home/multiomics/results/consensus_clustering/LNposvsLNneg/Ductal_noG2/',recursive = TRUE)
}
```

#### Proteome
##### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/batchcorrection/fullproteome/cycloess_rollup_genes_batchcorrected.tsv') %>% 
  drop_na()
design <- read_tsv(file = '/home/multiomics/results/design_files/design_full_ductal_noG2.tsv')
```

##### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
proteinDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.75)

proteinDataFiltered <- proteinDataFiltered %>% 
  dplyr::select(Protein,design$sample_id)

proteinDataReduced <- proteinDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
```

##### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/LNposvsLNneg/Ductal_noG2/')
results <-  ConsensusClusterPlus(d = proteinDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'proteome')
```

#### Phosphoproteome
##### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/batchcorrection/phosphoproteome/cycloess_peptide_flanking_batchcorrected.tsv') %>% 
  drop_na
design <- read_tsv(file = '/home/multiomics/results/design_files/design_phospho_ductal_noG2.tsv')
```

##### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
phosphoDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.5)

phosphoDataFiltered <- phosphoDataFiltered %>% 
  dplyr::select(Protein.Group,design$sample_id)

phosphoDataReduced <- phosphoDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix()
```

##### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/LNposvsLNneg/Ductal_noG2/')
results <-  ConsensusClusterPlus(d = phosphoDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'phosphoproteome')
```

#### Transcriptome
##### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/normalization/transcriptomics/genematrix_adjusted_cycloess.tsv') %>% 
  drop_na
design <- read_tsv(file = '/home/multiomics/results/design_files/design_RNA_ductal_noG2.tsv')
```

##### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
rnaDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.75)

rnaDataFiltered <- rnaDataFiltered %>% 
  dplyr::select(Gene.ID,design$sample_id)

rnaDataReduced <- rnaDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix()
```

##### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/LNposvsLNneg/Ductal_noG2/')
results <-  ConsensusClusterPlus(d = rnaDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'transcriptome')
```

### Lobular
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
if(!dir.exists(paths = '/home/multiomics/results/consensus_clustering/LNposvsLNneg/Lobular/')) {
  dir.create(path = '/home/multiomics/results/consensus_clustering/LNposvsLNneg/Lobular/',recursive = TRUE)
}
```

#### Proteome
##### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/batchcorrection/fullproteome/cycloess_rollup_genes_batchcorrected.tsv') %>% 
  drop_na()
design <- read_tsv(file = '/home/multiomics/results/design_files/design_full_lobular.tsv')
```

##### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
proteinDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.75)

proteinDataFiltered <- proteinDataFiltered %>% 
  dplyr::select(Protein,design$sample_id)

proteinDataReduced <- proteinDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
```

##### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/LNposvsLNneg/Lobular/')
results <-  ConsensusClusterPlus(d = proteinDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'proteome')
```

#### Phosphoproteome
##### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/batchcorrection/phosphoproteome/cycloess_peptide_flanking_batchcorrected.tsv') %>% 
  drop_na
design <- read_tsv(file = '/home/multiomics/results/design_files/design_phospho_lobular.tsv')
```
##### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
phosphoDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.5)

phosphoDataFiltered <- phosphoDataFiltered %>% 
  dplyr::select(Protein.Group,design$sample_id)

phosphoDataReduced <- phosphoDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix()
```

##### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/LNposvsLNneg/Lobular/')
results <-  ConsensusClusterPlus(d = phosphoDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'phosphoproteome')
```

#### Transcriptome
##### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/normalization/transcriptomics/genematrix_adjusted_cycloess.tsv') %>% 
  drop_na
design <- read_tsv(file = '/home/multiomics/results/design_files/design_RNA_lobular.tsv')
```

##### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
rnaDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.75)

rnaDataFiltered <- rnaDataFiltered %>% 
  dplyr::select(Gene.ID,design$sample_id)

rnaDataReduced <- rnaDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix()
```

##### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/LNposvsLNneg/Lobular/')
results <-  ConsensusClusterPlus(d = rnaDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'transcriptome')
```

## Group1 vs Group2
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
if(!dir.exists(paths = '/home/multiomics/results/consensus_clustering/Group1vsGroup2/')) {
  dir.create(path = '/home/multiomics/results/consensus_clustering/Group1vsGroup2/',recursive = TRUE)
}
```

### Proteome
#### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/batchcorrection/fullproteome/cycloess_rollup_genes_batchcorrected.tsv') %>% 
  drop_na()
design <- read_tsv(file = '/home/multiomics/results/design_files/design_full_g1g2.tsv')
```

#### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
proteinDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.75)

proteinDataFiltered <- proteinDataFiltered %>% 
  dplyr::select(Protein,design$sample_id)

proteinDataReduced <- proteinDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
```

#### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/Group1vsGroup2/')
results <-  ConsensusClusterPlus(d = proteinDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'proteome')
```

### Phosphoproteome
#### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/batchcorrection/phosphoproteome/cycloess_peptide_flanking_batchcorrected.tsv') %>% 
  drop_na
design <- read_tsv(file = '/home/multiomics/results/design_files/design_phospho_g1g2.tsv')
```

#### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
phosphoDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.5)

phosphoDataFiltered <- phosphoDataFiltered %>% 
  dplyr::select(Protein.Group,design$sample_id)

phosphoDataReduced <- phosphoDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix()
```

#### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/Group1vsGroup2/')
results <-  ConsensusClusterPlus(d = phosphoDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'phosphoproteome')
```

### Transcriptome
#### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/normalization/transcriptomics/genematrix_adjusted_cycloess.tsv') %>% 
  drop_na
design <- read_tsv(file = '/home/multiomics/results/design_files/design_RNA_g1g2.tsv')
```

#### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
rnaDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.75)

rnaDataFiltered <- rnaDataFiltered %>% 
  dplyr::select(Gene.ID,design$sample_id)

rnaDataReduced <- rnaDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix()
```

#### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/Group1vsGroup2/')
results <-  ConsensusClusterPlus(d = rnaDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'transcriptome')
```

## Group3 vs Group4
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
if(!dir.exists(paths = '/home/multiomics/results/consensus_clustering/Group3vsGroup4/')) {
  dir.create(path = '/home/multiomics/results/consensus_clustering/Group3vsGroup4/',recursive = TRUE)
}
```

### Proteome
#### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/batchcorrection/fullproteome/cycloess_rollup_genes_batchcorrected.tsv') %>% 
  drop_na()
design <- read_tsv(file = '/home/multiomics/results/design_files/design_full_g3g4.tsv')
```

#### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
proteinDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.75)

proteinDataFiltered <- proteinDataFiltered %>% 
  dplyr::select(Protein,design$sample_id)

proteinDataReduced <- proteinDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
```

#### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/Group3vsGroup4/')
results <-  ConsensusClusterPlus(d = proteinDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'proteome')
```

### Phosphoproteome
#### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/batchcorrection/phosphoproteome/cycloess_peptide_flanking_batchcorrected.tsv') %>% 
  drop_na
design <- read_tsv(file = '/home/multiomics/results/design_files/design_phospho_g3g4.tsv')
```

#### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
phosphoDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.5)

phosphoDataFiltered <- phosphoDataFiltered %>% 
  dplyr::select(Protein.Group,design$sample_id)

phosphoDataReduced <- phosphoDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix()
```

#### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/Group3vsGroup4/')
results <-  ConsensusClusterPlus(d = phosphoDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'phosphoproteome')
```

### Transcriptome
#### Load Data
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
data <- read_tsv(file = '/home/multiomics/results/normalization/transcriptomics/genematrix_adjusted_cycloess.tsv') %>% 
  drop_na
design <- read_tsv(file = '/home/multiomics/results/design_files/design_RNA_g3g4.tsv')
```

#### Preprocessing
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
rnaDataFiltered <- madFiltering(data = data,design = design,sampleCol = 'sample_id',cutoff = 0.75)

rnaDataFiltered <- rnaDataFiltered %>% 
  dplyr::select(Gene.ID,design$sample_id)

rnaDataReduced <- rnaDataFiltered %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix()
```

#### Plotting
```{r}
#| eval: !expr params$run
#| echo: !expr params$run
setwd('/home/multiomics/results/consensus_clustering/Group3vsGroup4/')
results <-  ConsensusClusterPlus(d = rnaDataReduced,
                                 maxK = 12,
                                 clusterAlg = 'pam', distance = 'pearson',
                                 seed = 123,
                                 verbose = TRUE,reps = 1000,plot = 'pdf',writeTable = TRUE,
                                 title = 'transcriptome')
```

:::