---
title: "Gene Set Enrichment Analysis - MOFA"
author: "Sergio Mosquim Junior"
format:
  html:
    toc: true
    code-fold: true
    embed-resources: true
    toc-location: left
    smooth-scroll: true
    message: false
engine: knitr
---
# Requires Packages:
```{r required packages}
#| message: false
#| warning: false
library(tidyverse)
library(clusterProfiler)
library(msigdbr)
library(MOFA2)
source(file = '/multiomics/notebooks/msigdb_gsea_function.R',chdir = TRUE)
```

# Gene Set Enrichment Analysis

## All samples with infiltration
### Model

```{r}
model <- readRDS(file = '/multiomics/results/MOFA/all_samples/infiltration/model.RDS')
```

### Rename Features

```{r}
features_names(model)[["proteomics"]] <- features_names(model)[["proteomics"]] %>% str_split_i(.,pattern='\\_',i=1)
features_names(model)[["phosphoproteomics"]] <- features_names(model)[["phosphoproteomics"]] %>% str_split_i(.,pattern='\\_',i=1)
features_names(model)[["transcriptomics"]] <- features_names(model)[["transcriptomics"]] %>% str_split_i(.,pattern='\\.',i=1)
```

### GSEA
```{r}
gsea <- gseaMsigMOFA(model = model,factor = 3,qvaluecutoff = 0.25)
```

### Plotting
```{r}
geneSetName <- msigdbr(species = 'Homo sapiens',category = 'H') %>% dplyr::select(gs_name) %>% unique(x = .)
resultsAll <- map(.x = names(gsea),.f = ~pluck(.x = gsea,.x,'result')) %>% 
  map(.f = ~left_join(x = geneSetName,y = .x,by = c('gs_name'='ID'))) %>% 
  map2(.y = c('Proteome','Phosphoproteome','Transcriptome'),.f = ~dplyr::mutate(.data=.x,Experiment=.y)) %>% 
  map(.f = ~dplyr::rename(.data=.x,ID=gs_name)) %>% 
  purrr::reduce(.f = dplyr::full_join)

resultsAll$ID <- resultsAll$ID %>% 
  str_remove(string = .,pattern = 'HALLMARK\\_') %>% 
  str_replace_all(.,'\\_',' ')

p <- ggplot(resultsAll, aes(Experiment, ID, fill= NES)) + 
  coord_fixed()+
  geom_tile(color='black',lwd=0.1,linetype=1) +
  scale_fill_gradient2(low="blue", high="red",na.value = 'white',mid = '#FFFFCC') + 
  theme_bw() +
  theme(axis.text.x = element_text(angle = 30,hjust = 1),
        panel.grid.major = element_blank(),
        legend.position = "right",
        legend.direction = "vertical")

if(!dir.exists(paths = '/multiomics/results/figures/MOFA/all_samples/infiltration/')) {
  dir.create(path = '/multiomics/results/figures/MOFA/all_samples/infiltration/',recursive = TRUE)
}
ggsave(filename = 'GSEA_allsamples_infiltration.png',plot = p,device = 'png',path = '/multiomics/results/figures/MOFA/all_samples/infiltration/',units = 'in',width = 10,height = 12,dpi = 600)

```


## Group1 vs Group2
### Model

```{r}
model <- readRDS(file = '/multiomics/results/MOFA/Group1vsGroup2/model.RDS')
```

### Rename Features

```{r}
features_names(model)[["proteomics"]] <- features_names(model)[["proteomics"]] %>% str_split_i(.,pattern='\\_',i=1)
features_names(model)[["phosphoproteomics"]] <- features_names(model)[["phosphoproteomics"]] %>% str_split_i(.,pattern='\\_',i=1)
features_names(model)[["transcriptomics"]] <- features_names(model)[["transcriptomics"]] %>% str_split_i(.,pattern='\\.',i=1)
```

### GSEA
```{r}
gsea <- gseaMsigMOFA(model = model,factor = 3,qvaluecutoff = 0.25)
```

### Plotting
```{r}
geneSetName <- msigdbr(species = 'Homo sapiens',category = 'H') %>% dplyr::select(gs_name) %>% unique(x = .)
resultsAll <- map(.x = names(gsea),.f = ~pluck(.x = gsea,.x,'result')) %>% 
  map(.f = ~left_join(x = geneSetName,y = .x,by = c('gs_name'='ID'))) %>% 
  map2(.y = c('Proteome','Phosphoproteome','Transcriptome'),.f = ~dplyr::mutate(.data=.x,Experiment=.y)) %>% 
  map(.f = ~dplyr::rename(.data=.x,ID=gs_name)) %>% 
  purrr::reduce(.f = dplyr::full_join)

resultsAll$ID <- resultsAll$ID %>% 
  str_remove(string = .,pattern = 'HALLMARK\\_') %>% 
  str_replace_all(.,'\\_',' ')

p <- ggplot(resultsAll, aes(Experiment, ID, fill= NES)) + 
  coord_fixed()+
  geom_tile(color='black',lwd=0.1,linetype=1) +
  scale_fill_gradient2(low="blue", high="red",na.value = 'white',mid = '#FFFFCC') + 
  theme_bw() +
  theme(axis.text.x = element_text(angle = 30,hjust = 1),
        panel.grid.major = element_blank(),
        legend.position = "right",
        legend.direction = "vertical")

if(!dir.exists(paths = '/multiomics/results/figures/MOFA/Group1vsGroup2/')) {
  dir.create(path = '/multiomics/results/figures/MOFA/Group1vsGroup2/',recursive = TRUE)
}
ggsave(filename = 'GSEA_Group1vsGroup2.png',plot = p,device = 'png',path = '/multiomics/results/figures/MOFA/Group1vsGroup2/',units = 'in',width = 10,height = 12,dpi = 600)

```
## Group3 vs Group4
### Model

```{r}
model <- readRDS(file = '/multiomics/results/MOFA/Group3vsGroup4/model.RDS')
```

### Rename Features

```{r}
features_names(model)[["proteomics"]] <- features_names(model)[["proteomics"]] %>% str_split_i(.,pattern='\\_',i=1)
features_names(model)[["phosphoproteomics"]] <- features_names(model)[["phosphoproteomics"]] %>% str_split_i(.,pattern='\\_',i=1)
features_names(model)[["transcriptomics"]] <- features_names(model)[["transcriptomics"]] %>% str_split_i(.,pattern='\\.',i=1)
```

### GSEA
```{r}
gsea <- gseaMsigMOFA(model = model,factor = 3,qvaluecutoff = 0.25)
```

### Plotting
```{r}
geneSetName <- msigdbr(species = 'Homo sapiens',category = 'H') %>% dplyr::select(gs_name) %>% unique(x = .)
resultsAll <- map(.x = names(gsea),.f = ~pluck(.x = gsea,.x,'result')) %>% 
  map(.f = ~left_join(x = geneSetName,y = .x,by = c('gs_name'='ID'))) %>% 
  map2(.y = c('Proteome','Phosphoproteome','Transcriptome'),.f = ~dplyr::mutate(.data=.x,Experiment=.y)) %>% 
  map(.f = ~dplyr::rename(.data=.x,ID=gs_name)) %>% 
  purrr::reduce(.f = dplyr::full_join)

resultsAll$ID <- resultsAll$ID %>% 
  str_remove(string = .,pattern = 'HALLMARK\\_') %>% 
  str_replace_all(.,'\\_',' ')

p <- ggplot(resultsAll, aes(Experiment, ID, fill= NES)) + 
  coord_fixed()+
  geom_tile(color='black',lwd=0.1,linetype=1) +
  scale_fill_gradient2(low="blue", high="red",na.value = 'white',mid = '#FFFFCC') + 
  theme_bw() +
  theme(axis.text.x = element_text(angle = 30,hjust = 1),
        panel.grid.major = element_blank(),
        legend.position = "right",
        legend.direction = "vertical")

if(!dir.exists(paths = '/multiomics/results/figures/MOFA/Group3vsGroup4/')) {
  dir.create(path = '/multiomics/results/figures/MOFA/Group3vsGroup4/',recursive = TRUE)
}
ggsave(filename = 'GSEA_Group3vsGroup4.png',plot = p,device = 'png',path = '/multiomics/results/figures/MOFA/Group3vsGroup4/',units = 'in',width = 10,height = 12,dpi = 600)

```