---
title: "Gene Set Enrichment Analysis"
author: "Sergio Mosquim Junior"
format:
  html:
    toc: true
    code-fold: true
    embed-resources: true
    toc-location: left
    smooth-scroll: true
    message: false
engine: knitr
---

# Requires Packages:
```{r required packages}
#| message: false
#| warning: false
library(tidyverse)
library(NormalyzerDE)
library(clusterProfiler)
library(msigdbr)
library(org.Hs.eg.db)
library(AnnotationDbi)
source(file = '~/multiomics/notebooks/msigdb_gsea_function.R',chdir = TRUE)
```

# Gene Set Enrichment Analysis
## Ductal vs Lobular
### Full Proteome
```{r}
data <- '~/multiomics/results/differentialexpression/DuctalvsLobular/fullproteome_protein/fullproteome_protein_stats.tsv'
gseaMsigProt <- gseaMsigDBProt(data = data,qvaluecutoff = 0.25,padjustmethod = 'fdr',species = 'Homo sapiens',category = 'H')
```

### Phosphoproteome
```{r}
data <- '~/multiomics/results/differentialexpression/DuctalvsLobular/phosphoproteome_peptide/phosphoproteome_peptide_stats.tsv'
gseaMsigPhospho <- gseaMsigDBPhospho(data = data,qvaluecutoff = 0.25,padjustmethod = 'fdr',species = 'Homo sapiens',category = 'H')
```

### Transcriptome
```{r}
data <- '~/multiomics/results/differentialexpression/DuctalvsLobular/transcriptomics_tpm_normalized/transcriptomics_tpm_normalized_stats.tsv'
gseaMsigRNA <- gseaMsigDBRNA(data = data,qvaluecutoff = 0.25,padjustmethod = 'fdr',species = 'Homo sapiens',category = 'H',genecol = 'Gene')
```

### MultiOmics GSEA Plot
```{r}
geneSetName <- msigdbr(species = 'Homo sapiens',category = 'H') %>% dplyr::select(gs_name) %>% unique(x = .)
resultsProt <- pluck(gseaMsigProt,'result') %>% 
  left_join(x=geneSetName,y = ., by = c('gs_name'='ID')) %>% 
  dplyr::mutate(Experiment = 'Proteome') %>% 
  dplyr::rename(ID=gs_name)
resultsPhospho <- pluck(gseaMsigPhospho,'result')%>% 
  left_join(x=geneSetName,y = ., by = c('gs_name'='ID')) %>% 
  dplyr::mutate(Experiment = 'Phosphoproteome') %>% 
  dplyr::rename(ID=gs_name)
resultsRNA <- pluck(gseaMsigRNA,'result') %>%
  left_join(x=geneSetName,y = ., by = c('gs_name'='ID')) %>% 
  dplyr::mutate(Experiment = 'Transcriptome') %>% 
  dplyr::rename(ID=gs_name)

allResults <- list(Proteomics=resultsProt,Phosphoproteomics=resultsPhospho,Transcriptomics=resultsRNA)
resultsAll <- purrr::reduce(.x = allResults,.f = dplyr::full_join)

resultsAll$ID <- resultsAll$ID %>% 
  str_remove(string = .,pattern = 'HALLMARK\\_') %>% 
  str_replace_all(.,'\\_',' ')

p <- ggplot(resultsAll, aes(Experiment, ID, fill= NES)) + 
  coord_fixed()+
  geom_tile(color='black',lwd=0.1,linetype=1) +
  scale_fill_gradient2(low="blue", high="red",na.value = 'white',mid = '#FFFFCC') + 
  theme_bw() +
  theme(axis.text.x = element_text(angle = 30,hjust = 1),
        panel.grid.major = element_blank(),
        legend.position = "right",
        legend.direction = "vertical")

if(!dir.exists(paths = '~/multiomics/results/figures')) {
  dir.create(path = '~/multiomics/results/figures',recursive = TRUE)
}
ggsave(filename = 'GSEA_ductalvslobular.png',plot = p,device = 'png',path = '~/multiomics/results/figures',units = 'in',width = 10,height = 12,dpi = 600)
```

## LN+ vs LN-
### Ductal
#### Full Proteome
```{r}
data <- '~/multiomics/results/differentialexpression/LNposvsLNneg/Ductal/fullproteome_protein/fullproteome_protein_stats.tsv'
gseaMsigProt <- gseaMsigDBProt(data = data,qvaluecutoff = 0.25,padjustmethod = 'fdr',species = 'Homo sapiens',category = 'H')
```

#### Phosphoproteome
```{r}
data <- '~/multiomics/results/differentialexpression/LNposvsLNneg/Ductal/phosphoproteome_peptide/phosphoproteome_peptide_stats.tsv'
gseaMsigPhospho <- gseaMsigDBPhospho(data = data,qvaluecutoff = 0.25,padjustmethod = 'fdr',species = 'Homo sapiens',category = 'H')
```

#### Transcriptome
```{r}
data <- '~/multiomics/results/differentialexpression/LNposvsLNneg/Ductal/transcriptomics_tpm_normalized/transcriptomics_tpm_normalized_stats.tsv'
gseaMsigRNA <- gseaMsigDBRNA(data = data,qvaluecutoff = 0.25,padjustmethod = 'fdr',species = 'Homo sapiens',category = 'H',genecol = 'Gene')
```

#### MultiOmics GSEA Plot
```{r}
resultsProt <- pluck(gseaMsigProt,'result') %>% 
  left_join(x=geneSetName,y = ., by = c('gs_name'='ID')) %>% 
  dplyr::mutate(Experiment = 'Proteome') %>% 
  dplyr::rename(ID=gs_name)
resultsPhospho <- pluck(gseaMsigPhospho,'result')%>% 
  left_join(x=geneSetName,y = ., by = c('gs_name'='ID')) %>% 
  dplyr::mutate(Experiment = 'Phosphoproteome') %>% 
  dplyr::rename(ID=gs_name)
resultsRNA <- pluck(gseaMsigRNA,'result') %>%
  left_join(x=geneSetName,y = ., by = c('gs_name'='ID')) %>% 
  dplyr::mutate(Experiment = 'Transcriptome') %>% 
  dplyr::rename(ID=gs_name)

allResults <- list(Proteomics=resultsProt,Phosphoproteomics=resultsPhospho,Transcriptomics=resultsRNA)
resultsAll <- purrr::reduce(.x = allResults,.f = dplyr::full_join)

resultsAll$ID <- resultsAll$ID %>% 
  str_remove(string = .,pattern = 'HALLMARK\\_') %>% 
  str_replace_all(.,'\\_',' ')

p <- ggplot(resultsAll, aes(Experiment, ID, fill= NES)) + 
  coord_fixed()+
  geom_tile(color='black',lwd=0.1,linetype=1) +
  scale_fill_gradient2(low="blue", high="red",na.value = 'white',mid = '#FFFFCC') + 
  theme_bw() +
  theme(axis.text.x = element_text(angle = 30,hjust = 1),
        panel.grid.major = element_blank(),
        legend.position = "right",
        legend.direction = "vertical")

if(!dir.exists(paths = '~/multiomics/results/figures')) {
  dir.create(path = '~/multiomics/results/figures',recursive = TRUE)
}
ggsave(filename = 'GSEA_LNposLNneg_ductal.png',plot = p,device = 'png',path = '~/multiomics/results/figures',units = 'in',width = 10,height = 12,dpi = 600)
```

### Lobular
#### Full Proteome
```{r}
data <- '~/multiomics/results/differentialexpression/LNposvsLNneg/Lobular/fullproteome_protein/fullproteome_protein_stats.tsv'
gseaMsigProt <- gseaMsigDBProt(data = data,qvaluecutoff = 0.25,padjustmethod = 'fdr',species = 'Homo sapiens',category = 'H')
```

#### Phosphoproteome
```{r}
data <- '~/multiomics/results/differentialexpression/LNposvsLNneg/Lobular/phosphoproteome_peptide/phosphoproteome_peptide_stats.tsv'
gseaMsigPhospho <- gseaMsigDBPhospho(data = data,qvaluecutoff = 0.25,padjustmethod = 'fdr',species = 'Homo sapiens',category = 'H')
```

#### Transcriptome
```{r}
data <- '~/multiomics/results/differentialexpression/LNposvsLNneg/Lobular/transcriptomics_tpm_normalized/transcriptomics_tpm_normalized_stats.tsv'
gseaMsigRNA <- gseaMsigDBRNA(data = data,qvaluecutoff = 0.25,padjustmethod = 'fdr',species = 'Homo sapiens',category = 'H',genecol = 'Gene')
```

#### MultiOmics GSEA Plot
```{r}
resultsProt <- pluck(gseaMsigProt,'result') %>% 
  left_join(x=geneSetName,y = ., by = c('gs_name'='ID')) %>% 
  dplyr::mutate(Experiment = 'Proteome') %>% 
  dplyr::rename(ID=gs_name)
resultsPhospho <- pluck(gseaMsigPhospho,'result')%>% 
  left_join(x=geneSetName,y = ., by = c('gs_name'='ID')) %>% 
  dplyr::mutate(Experiment = 'Phosphoproteome') %>% 
  dplyr::rename(ID=gs_name)
resultsRNA <- pluck(gseaMsigRNA,'result') %>%
  left_join(x=geneSetName,y = ., by = c('gs_name'='ID')) %>% 
  dplyr::mutate(Experiment = 'Transcriptome') %>% 
  dplyr::rename(ID=gs_name)

allResults <- list(Proteomics=resultsProt,Phosphoproteomics=resultsPhospho,Transcriptomics=resultsRNA)
resultsAll <- purrr::reduce(.x = allResults,.f = dplyr::full_join)

resultsAll$ID <- resultsAll$ID %>% 
  str_remove(string = .,pattern = 'HALLMARK\\_') %>% 
  str_replace_all(.,'\\_',' ')

p <- ggplot(resultsAll, aes(Experiment, ID, fill= NES)) + 
  coord_fixed()+
  geom_tile(color='black',lwd=0.1,linetype=1) +
  scale_fill_gradient2(low="blue", high="red",na.value = 'white',mid = '#FFFFCC') + 
  theme_bw() +
  theme(axis.text.x = element_text(angle = 30,hjust = 1),
        panel.grid.major = element_blank(),
        legend.position = "right",
        legend.direction = "vertical")

if(!dir.exists(paths = '~/multiomics/results/figures')) {
  dir.create(path = '~/multiomics/results/figures',recursive = TRUE)
}
ggsave(filename = 'GSEA_LNposLNneg_lobular.png',plot = p,device = 'png',path = '~/multiomics/results/figures',units = 'in',width = 10,height = 12,dpi = 600)
```

## Group1 vs Group2
### Full Proteome
```{r}
data <- '~/multiomics/results/differentialexpression/Group1vsGroup2/fullproteome_protein/fullproteome_protein_stats.tsv'
gseaMsigProt <- gseaMsigDBProt(data = data,qvaluecutoff = 0.25,padjustmethod = 'fdr',species = 'Homo sapiens',category = 'H')
```

### Phosphoproteome
```{r}
data <- '~/multiomics/results/differentialexpression/Group1vsGroup2/phosphoproteome_peptide/phosphoproteome_peptide_stats.tsv'
gseaMsigPhospho <- gseaMsigDBPhospho(data = data,qvaluecutoff = 0.25,padjustmethod = 'fdr',species = 'Homo sapiens',category = 'H')
```

### Transcriptome
```{r}
data <- '~/multiomics/results/differentialexpression/Group1vsGroup2/transcriptomics_tpm_normalized/transcriptomics_tpm_normalized_stats.tsv'
gseaMsigRNA <- gseaMsigDBRNA(data = data,qvaluecutoff = 0.25,padjustmethod = 'fdr',species = 'Homo sapiens',category = 'H',genecol = 'Gene')
```

### MultiOmics GSEA Plot
```{r}
resultsProt <- pluck(gseaMsigProt,'result') %>% 
  left_join(x=geneSetName,y = ., by = c('gs_name'='ID')) %>% 
  dplyr::mutate(Experiment = 'Proteome') %>% 
  dplyr::rename(ID=gs_name)
resultsPhospho <- pluck(gseaMsigPhospho,'result')%>% 
  left_join(x=geneSetName,y = ., by = c('gs_name'='ID')) %>% 
  dplyr::mutate(Experiment = 'Phosphoproteome') %>% 
  dplyr::rename(ID=gs_name)
resultsRNA <- pluck(gseaMsigRNA,'result') %>%
  left_join(x=geneSetName,y = ., by = c('gs_name'='ID')) %>% 
  dplyr::mutate(Experiment = 'Transcriptome') %>% 
  dplyr::rename(ID=gs_name)

allResults <- list(Proteomics=resultsProt,Phosphoproteomics=resultsPhospho,Transcriptomics=resultsRNA)
resultsAll <- purrr::reduce(.x = allResults,.f = dplyr::full_join)

resultsAll$ID <- resultsAll$ID %>% 
  str_remove(string = .,pattern = 'HALLMARK\\_') %>% 
  str_replace_all(.,'\\_',' ')

p <- ggplot(resultsAll, aes(Experiment, ID, fill= NES)) + 
  coord_fixed()+
  geom_tile(color='black',lwd=0.1,linetype=1) +
  scale_fill_gradient2(low="blue", high="red",na.value = 'white',mid = '#FFFFCC') + 
  theme_bw() +
  theme(axis.text.x = element_text(angle = 30,hjust = 1),
        panel.grid.major = element_blank(),
        legend.position = "right",
        legend.direction = "vertical")

if(!dir.exists(paths = '~/multiomics/results/figures')) {
  dir.create(path = '~/multiomics/results/figures',recursive = TRUE)
}
ggsave(filename = 'GSEA_Group1vsGroup2.png',plot = p,device = 'png',path = '~/multiomics/results/figures',units = 'in',width = 10,height = 12,dpi = 600)
```

## Group3 vs Group4
### Full Proteome
```{r}
data <- '~/multiomics/results/differentialexpression/Group3vsGroup4/fullproteome_protein/fullproteome_protein_stats.tsv'
gseaMsigProt <- gseaMsigDBProt(data = data,qvaluecutoff = 0.25,padjustmethod = 'fdr',species = 'Homo sapiens',category = 'H')
```

### Phosphoproteome
```{r}
data <- '~/multiomics/results/differentialexpression/Group3vsGroup4/phosphoproteome_peptide/phosphoproteome_peptide_stats.tsv'
gseaMsigPhospho <- gseaMsigDBPhospho(data = data,qvaluecutoff = 0.25,padjustmethod = 'fdr',species = 'Homo sapiens',category = 'H')
```

### Transcriptome
```{r}
data <- '~/multiomics/results/differentialexpression/Group3vsGroup4/transcriptomics_tpm_normalized/transcriptomics_tpm_normalized_stats.tsv'
gseaMsigRNA <- gseaMsigDBRNA(data = data,qvaluecutoff = 0.25,padjustmethod = 'fdr',species = 'Homo sapiens',category = 'H',genecol = 'Gene')
```

### MultiOmics GSEA Plot
```{r}
resultsProt <- pluck(gseaMsigProt,'result') %>% 
  left_join(x=geneSetName,y = ., by = c('gs_name'='ID')) %>% 
  dplyr::mutate(Experiment = 'Proteome') %>% 
  dplyr::rename(ID=gs_name)
resultsPhospho <- pluck(gseaMsigPhospho,'result')%>% 
  left_join(x=geneSetName,y = ., by = c('gs_name'='ID')) %>% 
  dplyr::mutate(Experiment = 'Phosphoproteome') %>% 
  dplyr::rename(ID=gs_name)
resultsRNA <- pluck(gseaMsigRNA,'result') %>%
  left_join(x=geneSetName,y = ., by = c('gs_name'='ID')) %>% 
  dplyr::mutate(Experiment = 'Transcriptome') %>% 
  dplyr::rename(ID=gs_name)

allResults <- list(Proteomics=resultsProt,Phosphoproteomics=resultsPhospho,Transcriptomics=resultsRNA)
resultsAll <- purrr::reduce(.x = allResults,.f = dplyr::full_join)

resultsAll$ID <- resultsAll$ID %>% 
  str_remove(string = .,pattern = 'HALLMARK\\_') %>% 
  str_replace_all(.,'\\_',' ')

p <- ggplot(resultsAll, aes(Experiment, ID, fill= NES)) + 
  coord_fixed()+
  geom_tile(color='black',lwd=0.1,linetype=1) +
  scale_fill_gradient2(low="blue", high="red",na.value = 'white',mid = '#FFFFCC') + 
  theme_bw() +
  theme(axis.text.x = element_text(angle = 30,hjust = 1),
        panel.grid.major = element_blank(),
        legend.position = "right",
        legend.direction = "vertical")

if(!dir.exists(paths = '~/multiomics/results/figures')) {
  dir.create(path = '~/multiomics/results/figures',recursive = TRUE)
}
ggsave(filename = 'GSEA_Group3vsGroup4.png',plot = p,device = 'png',path = '~/multiomics/results/figures',units = 'in',width = 10,height = 12,dpi = 600)
```