---
title: "MultiOmics Factor Analysis"
author: "Sergio Mosquim Junior"
format:
  html:
    toc: true
    code-fold: true
    embed-resources: true
    toc-location: left
    smooth-scroll: true
    message: false
engine: knitr
params:
  used_analysis: true
---

# Requires Packages:

```{r required packages}
#| message: false
#| warning: false
library(tidyverse)
library(MOFA2)
source(file = '/multiomics/notebooks/filtering.R',chdir = TRUE)
library(reticulate)
library(future)
library(future.callr)
library(furrr)
```

# MultiOmics Factor Analysis - MOFA
```{r}
#| eval: !expr (!params$used_analysis)
workers <- availableCores(constraints = 'connections',which = 'min')
plan(callr,workers=workers)
on.exit(expr = {plan(sequential)})
```

## All samples with infiltration
```{r}
#| eval: !expr (!params$used_analysis)
if(!dir.exists(paths = '/multiomics/results/figures/MOFA/all_samples/infiltration')) {
  dir.create(path = '/multiomics/results/figures/MOFA/all_samples/infiltration',recursive = TRUE)
}

if(!dir.exists(paths = '/multiomics/results/MOFA/all_samples/infiltration')) {
  dir.create(path = '/multiomics/results/MOFA/all_samples/infiltration',recursive = TRUE)
}
```

### Data preprocessing and multiomics dataframe creation
```{r}
#| eval: !expr (!params$used_analysis)
design <- read_tsv('/multiomics/results/design_files/design_multiomics_mofa.tsv')
proteindata <- read_tsv(file = '/multiomics/results/batchcorrection/fullproteome/cycloess_rollup_genes_batchcorrected.tsv')
prot <- proteindata %>% 
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Protein',cutoff = 0.75)
prot$Unique.Ids <- make.names(names = prot$Protein.Names,unique = TRUE)
protFeatures <- prot$Unique.Ids
prot <- prot %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
rownames(prot) <- protFeatures

phosphodata <- read_tsv(file = '/multiomics/results/batchcorrection/phosphoproteome/cycloess_peptide_flanking_batchcorrected.tsv')
phospho <- phosphodata %>%
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Modified.Sequence',cutoff = 0.5)
phospho$Unique.Ids <- make.names(names = phospho$Protein.Names,unique = TRUE)
phosphoFeatures <- phospho$Unique.Ids
phospho <- phospho %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
rownames(phospho) <- phosphoFeatures


genedata <- read_tsv(file = '/multiomics/data/transcriptomics/genematrix_adjusted_cycloess.tsv')
gene <- genedata %>% 
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Gene.ID',cutoff = 0.75)
gene$Unique.Ids <- make.names(names = gene$Gene.Name,unique = TRUE)
geneFeatures <- gene$Unique.Ids
gene <- gene %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix

immuneInfiltration <- read_tsv(file = '/multiomics/data/Immune_infiltration/combined_immuneinfiltration.tsv')
immuneFeat <- immuneInfiltration[['cell_type']]
immune <- immuneInfiltration %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix


dataMix <- list(proteomics=prot,
                phosphoproteomics=phospho,
                transcriptomics=gene,
                immuneinfiltration=immune)

response <- design %>% 
  dplyr::select(sample_id,Group.Info)
```

### Model Parameters
#### Sizes

```{r}
#| eval: !expr (!params$used_analysis)
purrr::map(.x = dataMix,.f = dim)
```

#### Create MOFA object

```{r}
#| eval: !expr (!params$used_analysis)
MOFAobject <- create_mofa(dataMix)
```

#### Plot data overview

```{r}
#| eval: !expr (!params$used_analysis)
plot_data_overview(MOFAobject)
```

#### Data options

```{r}
#| eval: !expr (!params$used_analysis)
data_opts <- get_default_data_options(MOFAobject)
data_opts$scale_views <- TRUE
```

#### Model options

```{r}
#| eval: !expr (!params$used_analysis)
model_opts <- get_default_model_options(MOFAobject)
model_opts$num_factors <- 20
model_opts$spikeslab_weights <- TRUE
model_opts$ard_weights <- TRUE
```

#### Training options

```{r}
#| eval: !expr (!params$used_analysis)
train_opts <- get_default_training_options(MOFAobject)
train_opts$convergence_mode <- "slow"
```

### Model Training

```{r}
#| eval: !expr (!params$used_analysis)
set.seed(123)
MOFAobject <- prepare_mofa(MOFAobject,data_options = data_opts,model_options = model_opts,training_options = train_opts)

trainedModels <- future_map(
  .options = furrr_options(seed = TRUE,globals = TRUE),
  .x = 1:10,
  .f = function(x){
    MOFAobject@training_options$seed <- (123+x)
    run_mofa(MOFAobject,use_basilisk = TRUE)
    }
  )

bestModel <- select_model(trainedModels)
```

### Model evaluation
#### Add metadata

```{r}
#| eval: !expr (!params$used_analysis)
# Add metadata
design <- design %>% 
  dplyr::rename(sample=sample_id)
samples_metadata(bestModel) <- design

# Add Feature Names
updated_features_names <- features_names(bestModel)
updated_features_names[["proteomics"]] <- protFeatures
updated_features_names[["phosphoproteomics"]] <- phosphoFeatures
updated_features_names[["transcriptomics"]] <- geneFeatures
updated_features_names[["immuneinfiltration"]] <- immuneFeat

features_names(bestModel) <- updated_features_names
```
```{r}
#| eval: !expr (!params$used_analysis)
slotNames(bestModel)
```

#### Save model
```{r}
#| eval: !expr (!params$used_analysis)
saveRDS(object = bestModel,file = '/multiomics/results/MOFA/all_samples/infiltration/model.RDS',compress = 'gzip')
```

#### Correlation between factors
```{r}
#| eval: !expr (!params$used_analysis)
factor_corr_plot <- {plot_factor_cor(object = bestModel)
  grDevices::recordPlot()}

ggsave(filename = 'factor_corr_plot.png',plot = grDevices::replayPlot(factor_corr_plot),device = 'png',path = '/multiomics/results/figures/MOFA/all_samples/infiltration/')
```

#### Variance decomposition
```{r}
#| eval: !expr (!params$used_analysis)
plotVariance <- plot_variance_explained(bestModel,max_r2 = 10)

ggsave(filename = 'MOFA_variance_allgroups.pdf',plot = plotVariance,device = 'pdf',path = '/multiomics/results/figures/MOFA/all_samples/infiltration/',dpi = 600,width = 10,height = 10,units = 'in')
```

#### Factor correlation
```{r}
#| eval: !expr (!params$used_analysis)
covariatePlot <- correlate_factors_with_covariates(bestModel, 
  covariates = c('Group.Info','Size.mm','NCN.PAM50','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","PR",'SSP.PAM50',"NHG",'BCFi_event','tumor_mutational_burden','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma"), 
  plot="log_pval",
  abs = TRUE,
  alpha = 0.05
)

ggsave(filename = 'covariance.pdf',plot = covariatePlot,device = 'pdf',path = '/multiomics/results/figures/MOFA/all_samples/infiltration/',dpi = 600,width = 10,height = 10,units = 'in')
```

#### Plot feature weights
```{r}
#| eval: !expr (!params$used_analysis)
immune_weights <- plot_top_weights(object = bestModel,
                                   view = 'immuneinfiltration',
                                   factors = 3,
                                   scale = TRUE,
                                   sign = 'all')

prot_weights <- plot_top_weights(bestModel,
                 view = "proteomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)

phospho_weigths <- plot_top_weights(bestModel,
                 view = "phosphoproteomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)
rna_weights <- plot_top_weights(bestModel,
                 view = "transcriptomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)

figPath <- c('prot_weights.png','phospho_weigths.png','rna_weights.png','immune_weights.png')
weightPlots <- list(prot_weights,phospho_weigths,rna_weights,immune_weights)
map2(.x = weightPlots,.y = figPath,.f = ~ggsave(filename = .y,plot = .x,device = 'png',path = '/multiomics/results/figures/MOFA/all_samples/infiltration/',dpi = 300,width = 15,height = 15,units = 'in'))
```

#### Plot Heatmaps
```{r}
#| eval: !expr (!params$used_analysis)
prot_heatmap <- plot_data_heatmap(bestModel, 
  view = "proteomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

rna_heatmap <- plot_data_heatmap(bestModel, 
  view = "transcriptomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

phospho_heatmap <- plot_data_heatmap(bestModel, 
  view = "phosphoproteomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

immune_heatmap <- plot_data_heatmap(bestModel, 
  view = "immuneinfiltration",
  factor = 3,  
  features = 8,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)


figPath <- c('prot_heatmap.png','phospho_heatmap.png','rna_heatmap.png','immune_heatmap')
heatmapPlots <- list(prot_heatmap,phospho_heatmap,rna_heatmap,immune_heatmap)
map2(.x = heatmapPlots,.y = figPath,.f = ~ggsave(filename = .y,plot = .x,device = 'png',path = '/multiomics/results/figures/MOFA/all_samples/infiltration/',dpi = 300,width = 15,height = 15,units = 'in'))
```

## All samples without infiltration
```{r}
#| eval: !expr (!params$used_analysis)
varskeep <- c('workers','madFiltering','NAFiltering','params')
rm(list = c(setdiff(ls(),varskeep)))
invisible(gc())

if(!dir.exists(paths = '/multiomics/results/figures/MOFA/all_samples/no_infiltration')) {
  dir.create(path = '/multiomics/results/figures/MOFA/all_samples/no_infiltration',recursive = TRUE)
}

if(!dir.exists(paths = '/multiomics/results/MOFA/all_samples/no_infiltration')) {
  dir.create(path = '/multiomics/results/MOFA/all_samples/no_infiltration',recursive = TRUE)
}
```

### Data preprocessing and multiomics dataframe creation
```{r}
#| eval: !expr (!params$used_analysis)
design <- read_tsv('/multiomics/results/design_files/design_multiomics_mofa.tsv')
proteindata <- read_tsv(file = '/multiomics/results/batchcorrection/fullproteome/cycloess_rollup_genes_batchcorrected.tsv')
prot <- proteindata %>% 
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Protein',cutoff = 0.75)
prot$Unique.Ids <- make.names(names = prot$Protein.Names,unique = TRUE)
protFeatures <- prot$Unique.Ids
prot <- prot %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
rownames(prot) <- protFeatures

phosphodata <- read_tsv(file = '/multiomics/results/batchcorrection/phosphoproteome/cycloess_peptide_flanking_batchcorrected.tsv')
phospho <- phosphodata %>%
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Modified.Sequence',cutoff = 0.5)
phospho$Unique.Ids <- make.names(names = phospho$Protein.Names,unique = TRUE)
phosphoFeatures <- phospho$Unique.Ids
phospho <- phospho %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
rownames(phospho) <- phosphoFeatures

genedata <- read_tsv(file = '/multiomics/data/transcriptomics/genematrix_adjusted_cycloess.tsv')
gene <- genedata %>% 
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Gene.ID',cutoff = 0.75)
gene$Unique.Ids <- make.names(names = gene$Gene.Name,unique = TRUE)
geneFeatures <- gene$Unique.Ids
gene <- gene %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix

dataMix <- list(proteomics=prot,
                phosphoproteomics=phospho,
                transcriptomics=gene)

response <- design %>% 
  dplyr::select(sample_id,Group.Info)
```

### Model Parameters
#### Sizes

```{r}
#| eval: !expr (!params$used_analysis)
purrr::map(.x = dataMix,.f = dim)
```

#### Create MOFA object

```{r}
#| eval: !expr (!params$used_analysis)
MOFAobject <- create_mofa(dataMix)
```

#### Plot data overview

```{r}
#| eval: !expr (!params$used_analysis)
plot_data_overview(MOFAobject)
```

#### Data options

```{r}
#| eval: !expr (!params$used_analysis)
data_opts <- get_default_data_options(MOFAobject)
data_opts$scale_views <- TRUE
```

#### Model options

```{r}
#| eval: !expr (!params$used_analysis)
model_opts <- get_default_model_options(MOFAobject)
model_opts$num_factors <- 20
model_opts$spikeslab_weights <- TRUE
model_opts$ard_weights <- TRUE
```

#### Training options

```{r}
#| eval: !expr (!params$used_analysis)
train_opts <- get_default_training_options(MOFAobject)
train_opts$convergence_mode <- "slow"
```

### Model Training

```{r}
#| eval: !expr (!params$used_analysis)
set.seed(123)
MOFAobject <- prepare_mofa(MOFAobject,data_options = data_opts,model_options = model_opts,training_options = train_opts)

trainedModels <- future_map(
  .options = furrr_options(seed = TRUE,globals = TRUE),
  .x = 1:10,
  .f = function(x){
    MOFAobject@training_options$seed <- (123+x)
    run_mofa(MOFAobject,use_basilisk = TRUE)
    }
  )

bestModel <- select_model(trainedModels)
```

### Model evaluation
#### Add metadata

```{r}
#| eval: !expr (!params$used_analysis)
# Add metadata
design <- design %>% 
  dplyr::rename(sample=sample_id)
samples_metadata(bestModel) <- design

# Add Feature Names
updated_features_names <- features_names(bestModel)
updated_features_names[["proteomics"]] <- protFeatures
updated_features_names[["phosphoproteomics"]] <- phosphoFeatures
updated_features_names[["transcriptomics"]] <- geneFeatures

features_names(bestModel) <- updated_features_names
```
```{r}
#| eval: !expr (!params$used_analysis)
slotNames(bestModel)
```

#### Save model
```{r}
#| eval: !expr (!params$used_analysis)
saveRDS(object = bestModel,file = '/multiomics/results/MOFA/all_samples/no_infiltration/model.RDS',compress = 'gzip')
```

#### Correlation between factors
```{r}
#| eval: !expr (!params$used_analysis)
factor_corr_plot <- {plot_factor_cor(bestModel)
  grDevices::recordPlot()}

ggsave(filename = 'factor_corr_plot.png',plot = grDevices::replayPlot(factor_corr_plot),device = 'png',path = '/multiomics/results/figures/MOFA/all_samples/no_infiltration/')
```

#### Variance decomposition
```{r}
#| eval: !expr (!params$used_analysis)
plotVariance <- plot_variance_explained(bestModel,max_r2 = 10)

ggsave(filename = 'MOFA_variance_allgroups.pdf',plot = plotVariance,device = 'pdf',path = '/multiomics/results/figures/MOFA/all_samples/no_infiltration/',dpi = 600,width = 10,height = 10,units = 'in')
```

#### Factor correlation
```{r}
#| eval: !expr (!params$used_analysis)
covariatePlot <- correlate_factors_with_covariates(bestModel, 
  covariates = c('Group.Info','Size.mm','NCN.PAM50','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","PR",'SSP.PAM50',"NHG",'BCFi_event','tumor_mutational_burden','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma"), 
  plot="log_pval",
  abs = TRUE,
  alpha = 0.05
)

ggsave(filename = 'covariance.pdf',plot = covariatePlot,device = 'pdf',path = '/multiomics/results/figures/MOFA/all_samples/no_infiltration/',dpi = 600,width = 10,height = 10,units = 'in')
```

#### Plot feature weights
```{r}
#| eval: !expr (!params$used_analysis)
prot_weights <- plot_top_weights(bestModel,
                 view = "proteomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)

phospho_weigths <- plot_top_weights(bestModel,
                 view = "phosphoproteomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)
rna_weights <- plot_top_weights(bestModel,
                 view = "transcriptomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)

figPath <- c('prot_weights.png','phospho_weigths.png','rna_weights.png')
weightPlots <- list(prot_weights,phospho_weigths,rna_weights)
map2(.x = weightPlots,.y = figPath,.f = ~ggsave(filename = .y,plot = .x,device = 'png',path = '/multiomics/results/figures/MOFA/all_samples/no_infiltration/',dpi = 300,width = 15,height = 15,units = 'in'))
```

#### Plot Heatmaps
```{r}
#| eval: !expr (!params$used_analysis)
prot_heatmap <- plot_data_heatmap(bestModel, 
  view = "proteomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

rna_heatmap <- plot_data_heatmap(bestModel, 
  view = "transcriptomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

phospho_heatmap <- plot_data_heatmap(bestModel, 
  view = "phosphoproteomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

figPath <- c('prot_heatmap.png','phospho_heatmap.png','rna_heatmap.png')
heatmapPlots <- list(prot_heatmap,phospho_heatmap,rna_heatmap)
map2(.x = heatmapPlots,.y = figPath,.f = ~ggsave(filename = .y,plot = .x,device = 'png',path = '/multiomics/results/figures/MOFA/all_samples/no_infiltration/',dpi = 300,width = 15,height = 15,units = 'in'))
```

## Group1 vs Group2
```{r}
#| eval: !expr (params$used_analysis)
varskeep <- c('workers','madFiltering','NAFiltering','params')
rm(list = c(setdiff(ls(),varskeep)))
invisible(gc())

if(!dir.exists(paths = '/multiomics/results/figures/MOFA/Group1vsGroup2/')) {
  dir.create(path = '/multiomics/results/figures/MOFA/Group1vsGroup2/',recursive = TRUE)
}

if(!dir.exists(paths = '/multiomics/results/MOFA/Group1vsGroup2/')) {
  dir.create(path = '/multiomics/results/MOFA/Group1vsGroup2/',recursive = TRUE)
}
```

### Data preprocessing and multiomics dataframe creation
```{r}
#| eval: !expr (params$used_analysis)
design <- read_tsv('/multiomics/results/design_files/design_multiomics_mofa.tsv') %>% 
    filter(str_detect(string = Group.Info,pattern = 'Group1|Group2'))
proteindata <- read_tsv(file = '/multiomics/results/batchcorrection/fullproteome/cycloess_rollup_genes_batchcorrected.tsv')
prot <- proteindata %>% 
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Protein',cutoff = 0.75)
prot$Unique.Ids <- make.names(names = prot$Protein.Names,unique = TRUE)
protFeatures <- prot$Unique.Ids
protIds <- prot %>% 
  select(matches("Name|Gene|Ids|Sequence|Phosphosite|Residue"))
prot <- prot %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
rownames(prot) <- protFeatures

phosphodata <- read_tsv(file = '/multiomics/results/batchcorrection/phosphoproteome/cycloess_peptide_flanking_batchcorrected.tsv')
phospho <- phosphodata %>%
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Modified.Sequence',cutoff = 0.5)
phospho$Unique.Ids <- make.names(names = phospho$Protein.Names,unique = TRUE)
phosphoFeatures <- phospho$Unique.Ids
phosphoIds <- phospho %>% 
  select(matches("Name|Gene|Ids|Sequence|Phosphosite|Residue"))
phospho <- phospho %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
rownames(phospho) <- phosphoFeatures

genedata <- read_tsv(file = '/multiomics/data/transcriptomics/genematrix_adjusted_cycloess.tsv')
gene <- genedata %>% 
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Gene.ID',cutoff = 0.75)
gene$Unique.Ids <- make.names(names = gene$Gene.Name,unique = TRUE)
geneFeatures <- gene$Unique.Ids
geneIds <- gene %>% 
  select(matches("Name|Gene|Ids|Sequence|Phosphosite|Residue"))
gene <- gene %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix

immuneInfiltration <- read_tsv(file = '/multiomics/data/Immune_infiltration/combined_immuneinfiltration.tsv')
immuneFeat <- immuneInfiltration[['cell_type']]
immune <- immuneInfiltration %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix


dataMix <- list(proteomics=prot,
                phosphoproteomics=phospho,
                transcriptomics=gene,
                immuneinfiltration=immune)

response <- design %>% 
  dplyr::select(sample_id,Group.Info)

allIds <- list(proteomics=protIds,phosphoproteomics=phosphoIds,transcriptomics=geneIds) %>% 
  map2(.y = c('proteomics','phosphoproteomics','transcriptomics'),.f = function(x,y){
    data <- x %>% mutate(across(Unique.Ids,~paste0(.x,'.',y)))
  }) %>% 
  list_rbind()

write_tsv(x = allIds,file = '/multiomics/results/MOFA/Group1vsGroup2/featIDs.tsv')
```

### Model Parameters
#### Sizes

```{r}
#| eval: !expr (params$used_analysis)
purrr::map(.x = dataMix,.f = dim)
```

#### Create MOFA object

```{r}
#| eval: !expr (params$used_analysis)
MOFAobject <- create_mofa(dataMix)
```

#### Plot data overview

```{r}
#| eval: !expr (params$used_analysis)
plot_data_overview(MOFAobject)
```

#### Data options

```{r}
#| eval: !expr (params$used_analysis)
data_opts <- get_default_data_options(MOFAobject)
data_opts$scale_views <- TRUE
```

#### Model options

```{r}
#| eval: !expr (params$used_analysis)
model_opts <- get_default_model_options(MOFAobject)
model_opts$num_factors <- 20
model_opts$spikeslab_weights <- TRUE
model_opts$ard_weights <- TRUE
```

#### Training options

```{r}
#| eval: !expr (params$used_analysis)
train_opts <- get_default_training_options(MOFAobject)
train_opts$convergence_mode <- "slow"
```

### Model Training

```{r}
#| eval: !expr (params$used_analysis)
set.seed(123)
MOFAobject <- prepare_mofa(MOFAobject,data_options = data_opts,model_options = model_opts,training_options = train_opts)

trainedModels <- future_map(
  .options = furrr_options(seed = TRUE,globals = TRUE),
  .x = 1:10,
  .f = function(x){
    MOFAobject@training_options$seed <- (123+x)
    run_mofa(MOFAobject,use_basilisk = TRUE)
    }
  )

bestModel <- select_model(trainedModels)
```

### Model evaluation
#### Add metadata

```{r}
#| eval: !expr (params$used_analysis)
# Add metadata
design <- design %>% 
  dplyr::rename(sample=sample_id)
samples_metadata(bestModel) <- design

# Add Feature Names
updated_features_names <- features_names(bestModel)
updated_features_names[["proteomics"]] <- protFeatures
updated_features_names[["phosphoproteomics"]] <- phosphoFeatures
updated_features_names[["transcriptomics"]] <- geneFeatures
updated_features_names[["immuneinfiltration"]] <- immuneFeat


features_names(bestModel) <- updated_features_names
```
```{r}
#| eval: !expr (params$used_analysis)
slotNames(bestModel)
```

#### Save model
```{r}
#| eval: !expr (params$used_analysis)
saveRDS(object = bestModel,file = '/multiomics/results/MOFA/Group1vsGroup2/model.RDS',compress = 'gzip')
```

#### Correlation between factors
```{r}
#| eval: !expr (params$used_analysis)
factor_corr_plot <- {plot_factor_cor(object = bestModel)
  grDevices::recordPlot()}

ggsave(filename = 'factor_corr_plot.png',plot = grDevices::replayPlot(factor_corr_plot),device = 'png',path = '/multiomics/results/figures/MOFA/Group1vsGroup2/')
```

#### Variance decomposition
```{r}
#| eval: !expr (params$used_analysis)
plotVariance <- plot_variance_explained(bestModel,max_r2 = 10)

ggsave(filename = 'MOFA_variance_allgroups.pdf',plot = plotVariance,device = 'pdf',path = '/multiomics/results/figures/MOFA/Group1vsGroup2/',dpi = 600,width = 10,height = 10,units = 'in')
```

#### Factor correlation
```{r}
#| eval: !expr (params$used_analysis)
covariatePlot <- correlate_factors_with_covariates(bestModel, 
  covariates = c('Group.Info','Size.mm','NCN.PAM50','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","PR",'SSP.PAM50',"NHG",'BCFi_event','tumor_mutational_burden',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma"), 
  plot="log_pval",
  abs = TRUE,
  alpha = 0.05
)

ggsave(filename = 'covariance.pdf',plot = covariatePlot,device = 'pdf',path = '/multiomics/results/figures/MOFA/Group1vsGroup2/',dpi = 600,width = 10,height = 10,units = 'in')
```

#### Plot feature weights
```{r}
#| eval: !expr (params$used_analysis)
immune_weights <- plot_top_weights(object = bestModel,
                                   view = 'immuneinfiltration',
                                   factors = 3,
                                   scale = TRUE,
                                   sign = 'all')

prot_weights <- plot_top_weights(bestModel,
                 view = "proteomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)

phospho_weigths <- plot_top_weights(bestModel,
                 view = "phosphoproteomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)
rna_weights <- plot_top_weights(bestModel,
                 view = "transcriptomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)

figPath <- c('prot_weights.png','phospho_weigths.png','rna_weights.png','immune_weights.png')
weightPlots <- list(prot_weights,phospho_weigths,rna_weights,immune_weights)
map2(.x = weightPlots,.y = figPath,.f = ~ggsave(filename = .y,plot = .x,device = 'png',path = '/multiomics/results/figures/MOFA/Group1vsGroup2/',dpi = 300,width = 15,height = 15,units = 'in'))
```

#### Plot Heatmaps
```{r}
#| eval: !expr (params$used_analysis)
prot_heatmap <- plot_data_heatmap(bestModel, 
  view = "proteomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

rna_heatmap <- plot_data_heatmap(bestModel, 
  view = "transcriptomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

phospho_heatmap <- plot_data_heatmap(bestModel, 
  view = "phosphoproteomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

immune_heatmap <- plot_data_heatmap(bestModel, 
  view = "immuneinfiltration",
  factor = 3,  
  features = 8,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)


figPath <- c('prot_heatmap.png','phospho_heatmap.png','rna_heatmap.png','immune_heatmap')
heatmapPlots <- list(prot_heatmap,phospho_heatmap,rna_heatmap,immune_heatmap)
map2(.x = heatmapPlots,.y = figPath,.f = ~ggsave(filename = .y,plot = .x,device = 'png',path = '/multiomics/results/figures/MOFA/Group1vsGroup2/',dpi = 300,width = 15,height = 15,units = 'in'))
```

## Group3 vs Group4
```{r}
#| eval: !expr (!params$used_analysis)
varskeep <- c('workers','madFiltering','NAFiltering','params')
rm(list = c(setdiff(ls(),varskeep)))
invisible(gc())

if(!dir.exists(paths = '/multiomics/results/figures/MOFA/Group3vsGroup4/')) {
  dir.create(path = '/multiomics/results/figures/MOFA/Group3vsGroup4/',recursive = TRUE)
}

if(!dir.exists(paths = '/multiomics/results/MOFA/Group3vsGroup4/')) {
  dir.create(path = '/multiomics/results/MOFA/Group3vsGroup4/',recursive = TRUE)
}
```

### Data preprocessing and multiomics dataframe creation
```{r}
#| eval: !expr (!params$used_analysis)
design <- read_tsv('/multiomics/results/design_files/design_multiomics_mofa.tsv') %>% 
    filter(str_detect(string = Group.Info,pattern = 'Group3|Group4'))
proteindata <- read_tsv(file = '/multiomics/results/batchcorrection/fullproteome/cycloess_rollup_genes_batchcorrected.tsv')
prot <- proteindata %>% 
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Protein',cutoff = 0.75)
prot$Unique.Ids <- make.names(names = prot$Protein.Names,unique = TRUE)
protFeatures <- prot$Unique.Ids
prot <- prot %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
rownames(prot) <- protFeatures

phosphodata <- read_tsv(file = '/multiomics/results/batchcorrection/phosphoproteome/cycloess_peptide_flanking_batchcorrected.tsv')
phospho <- phosphodata %>%
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Modified.Sequence',cutoff = 0.5)
phospho$Unique.Ids <- make.names(names = phospho$Protein.Names,unique = TRUE)
phosphoFeatures <- phospho$Unique.Ids
phospho <- phospho %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
rownames(phospho) <- phosphoFeatures

genedata <- read_tsv(file = '/multiomics/data/transcriptomics/genematrix_adjusted_cycloess.tsv')
gene <- genedata %>% 
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Gene.ID',cutoff = 0.75)
gene$Unique.Ids <- make.names(names = gene$Gene.Name,unique = TRUE)
geneFeatures <- gene$Unique.Ids
gene <- gene %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix

immuneInfiltration <- read_tsv(file = '/multiomics/data/Immune_infiltration/combined_immuneinfiltration.tsv')
immuneFeat <- immuneInfiltration[['cell_type']]
immune <- immuneInfiltration %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix


dataMix <- list(proteomics=prot,
                phosphoproteomics=phospho,
                transcriptomics=gene,
                immuneinfiltration=immune)

response <- design %>% 
  dplyr::select(sample_id,Group.Info)
```

### Model Parameters
#### Sizes

```{r}
#| eval: !expr (!params$used_analysis)
purrr::map(.x = dataMix,.f = dim)
```

#### Create MOFA object

```{r}
#| eval: !expr (!params$used_analysis)
MOFAobject <- create_mofa(dataMix)
```

#### Plot data overview

```{r}
#| eval: !expr (!params$used_analysis)
plot_data_overview(MOFAobject)
```

#### Data options

```{r}
#| eval: !expr (!params$used_analysis)
data_opts <- get_default_data_options(MOFAobject)
data_opts$scale_views <- TRUE
```

#### Model options

```{r}
#| eval: !expr (!params$used_analysis)
model_opts <- get_default_model_options(MOFAobject)
model_opts$num_factors <- 20
model_opts$spikeslab_weights <- TRUE
model_opts$ard_weights <- TRUE
```

#### Training options

```{r}
#| eval: !expr (!params$used_analysis)
train_opts <- get_default_training_options(MOFAobject)
train_opts$convergence_mode <- "slow"
```

### Model Training

```{r}
#| eval: !expr (!params$used_analysis)
set.seed(123)
MOFAobject <- prepare_mofa(MOFAobject,data_options = data_opts,model_options = model_opts,training_options = train_opts)

trainedModels <- future_map(
  .options = furrr_options(seed = TRUE,globals = TRUE),
  .x = 1:10,
  .f = function(x){
    MOFAobject@training_options$seed <- (123+x)
    run_mofa(MOFAobject,use_basilisk = TRUE)
    }
  )

bestModel <- select_model(trainedModels)
```

### Model evaluation
#### Add metadata

```{r}
#| eval: !expr (!params$used_analysis)
# Add metadata
design <- design %>% 
  dplyr::rename(sample=sample_id)
samples_metadata(bestModel) <- design

# Add Feature Names
updated_features_names <- features_names(bestModel)
updated_features_names[["proteomics"]] <- protFeatures
updated_features_names[["phosphoproteomics"]] <- phosphoFeatures
updated_features_names[["transcriptomics"]] <- geneFeatures
updated_features_names[["immuneinfiltration"]] <- immuneFeat

features_names(bestModel) <- updated_features_names
```
```{r}
#| eval: !expr (!params$used_analysis)
slotNames(bestModel)
```

#### Save model
```{r}
#| eval: !expr (!params$used_analysis)
saveRDS(object = bestModel,file = '/multiomics/results/MOFA/Group3vsGroup4/model.RDS',compress = 'gzip')
```

#### Correlation between factors
```{r}
#| eval: !expr (!params$used_analysis)
factor_corr_plot <- {plot_factor_cor(object = bestModel)
  grDevices::recordPlot()}

ggsave(filename = 'factor_corr_plot.png',plot = grDevices::replayPlot(factor_corr_plot),device = 'png',path = '/multiomics/results/figures/MOFA/Group3vsGroup4/')
```

#### Variance decomposition
```{r}
#| eval: !expr (!params$used_analysis)
plotVariance <- plot_variance_explained(bestModel,max_r2 = 10)

ggsave(filename = 'MOFA_variance_allgroups.pdf',plot = plotVariance,device = 'pdf',path = '/multiomics/results/figures/MOFA/Group3vsGroup4/',dpi = 600,width = 10,height = 10,units = 'in')
```

#### Factor correlation
```{r}
#| eval: !expr (!params$used_analysis)
covariatePlot <- correlate_factors_with_covariates(bestModel, 
  covariates = c('Group.Info','Size.mm','NCN.PAM50','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","PR",'SSP.PAM50',"NHG",'BCFi_event','tumor_mutational_burden','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma"), 
  plot="log_pval",
  abs = TRUE,
  alpha = 0.05
)

ggsave(filename = 'covariance.pdf',plot = covariatePlot,device = 'pdf',path = '/multiomics/results/figures/MOFA/Group3vsGroup4/',dpi = 600,width = 10,height = 10,units = 'in')
```

#### Plot feature weights
```{r}
#| eval: !expr (!params$used_analysis)
immune_weights <- plot_top_weights(object = bestModel,
                                   view = 'immuneinfiltration',
                                   factors = 3,
                                   scale = TRUE,
                                   sign = 'all')

prot_weights <- plot_top_weights(bestModel,
                 view = "proteomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)

phospho_weigths <- plot_top_weights(bestModel,
                 view = "phosphoproteomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)
rna_weights <- plot_top_weights(bestModel,
                 view = "transcriptomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)

figPath <- c('prot_weights.png','phospho_weigths.png','rna_weights.png','immune_weights.png')
weightPlots <- list(prot_weights,phospho_weigths,rna_weights,immune_weights)
map2(.x = weightPlots,.y = figPath,.f = ~ggsave(filename = .y,plot = .x,device = 'png',path = '/multiomics/results/figures/MOFA/Group3vsGroup4/',dpi = 300,width = 15,height = 15,units = 'in'))
```

#### Plot Heatmaps
```{r}
#| eval: !expr (!params$used_analysis)
prot_heatmap <- plot_data_heatmap(bestModel, 
  view = "proteomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

rna_heatmap <- plot_data_heatmap(bestModel, 
  view = "transcriptomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

phospho_heatmap <- plot_data_heatmap(bestModel, 
  view = "phosphoproteomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

immune_heatmap <- plot_data_heatmap(bestModel, 
  view = "immuneinfiltration",
  factor = 3,  
  features = 8,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)


figPath <- c('prot_heatmap.png','phospho_heatmap.png','rna_heatmap.png','immune_heatmap')
heatmapPlots <- list(prot_heatmap,phospho_heatmap,rna_heatmap,immune_heatmap)
map2(.x = heatmapPlots,.y = figPath,.f = ~ggsave(filename = .y,plot = .x,device = 'png',path = '/multiomics/results/figures/MOFA/Group3vsGroup4/',dpi = 300,width = 15,height = 15,units = 'in'))
```

## Lymphnode Positive vs Negative - Ductal
```{r}
#| eval: !expr (!params$used_analysis)
varskeep <- c('workers','madFiltering','NAFiltering','params')
rm(list = c(setdiff(ls(),varskeep)))
invisible(gc())

if(!dir.exists(paths = '/multiomics/results/figures/MOFA/LNposvsLNneg/Ductal/')) {
  dir.create(path = '/multiomics/results/figures/MOFA/LNposvsLNneg/Ductal/',recursive = TRUE)
}

if(!dir.exists(paths = '/multiomics/results/MOFA/LNposvsLNneg/Ductal/')) {
  dir.create(path = '/multiomics/results/MOFA/LNposvsLNneg/Ductal/',recursive = TRUE)
}
```

### Data preprocessing and multiomics dataframe creation
```{r}
#| eval: !expr (!params$used_analysis)
design <- read_tsv('/multiomics/results/design_files/design_multiomics_mofa.tsv') %>% 
    filter(str_detect(string = InvCa.type,pattern = 'Ductal'))
proteindata <- read_tsv(file = '/multiomics/results/batchcorrection/fullproteome/cycloess_rollup_genes_batchcorrected.tsv')
prot <- proteindata %>% 
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Protein',cutoff = 0.75)
prot$Unique.Ids <- make.names(names = prot$Protein.Names,unique = TRUE)
protFeatures <- prot$Unique.Ids
prot <- prot %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
rownames(prot) <- protFeatures

phosphodata <- read_tsv(file = '/multiomics/results/batchcorrection/phosphoproteome/cycloess_peptide_flanking_batchcorrected.tsv')
phospho <- phosphodata %>%
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Modified.Sequence',cutoff = 0.5)
phospho$Unique.Ids <- make.names(names = phospho$Protein.Names,unique = TRUE)
phosphoFeatures <- phospho$Unique.Ids
phospho <- phospho %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
rownames(phospho) <- phosphoFeatures

genedata <- read_tsv(file = '/multiomics/data/transcriptomics/genematrix_adjusted_cycloess.tsv')
gene <- genedata %>% 
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Gene.ID',cutoff = 0.75)
gene$Unique.Ids <- make.names(names = gene$Gene.Name,unique = TRUE)
geneFeatures <- gene$Unique.Ids
gene <- gene %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix

immuneInfiltration <- read_tsv(file = '/multiomics/data/Immune_infiltration/combined_immuneinfiltration.tsv')
immuneFeat <- immuneInfiltration[['cell_type']]
immune <- immuneInfiltration %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix


dataMix <- list(proteomics=prot,
                phosphoproteomics=phospho,
                transcriptomics=gene,
                immuneinfiltration=immune)

response <- design %>% 
  dplyr::select(sample_id,LN)
```

### Model Parameters
#### Sizes

```{r}
#| eval: !expr (!params$used_analysis)
purrr::map(.x = dataMix,.f = dim)
```

#### Create MOFA object

```{r}
#| eval: !expr (!params$used_analysis)
MOFAobject <- create_mofa(dataMix)
```

#### Plot data overview

```{r}
#| eval: !expr (!params$used_analysis)
plot_data_overview(MOFAobject)
```

#### Data options

```{r}
#| eval: !expr (!params$used_analysis)
data_opts <- get_default_data_options(MOFAobject)
data_opts$scale_views <- TRUE
```

#### Model options

```{r}
#| eval: !expr (!params$used_analysis)
model_opts <- get_default_model_options(MOFAobject)
model_opts$num_factors <- 20
model_opts$spikeslab_weights <- TRUE
model_opts$ard_weights <- TRUE
```

#### Training options

```{r}
#| eval: !expr (!params$used_analysis)
train_opts <- get_default_training_options(MOFAobject)
train_opts$convergence_mode <- "slow"
```

### Model Training

```{r}
#| eval: !expr (!params$used_analysis)
set.seed(123)
MOFAobject <- prepare_mofa(MOFAobject,data_options = data_opts,model_options = model_opts,training_options = train_opts)

trainedModels <- future_map(
  .options = furrr_options(seed = TRUE,globals = TRUE),
  .x = 1:10,
  .f = function(x){
    MOFAobject@training_options$seed <- (123+x)
    run_mofa(MOFAobject,use_basilisk = TRUE)
    }
  )

bestModel <- select_model(trainedModels)
```

### Model evaluation
#### Add metadata

```{r}
#| eval: !expr (!params$used_analysis)
# Add metadata
design <- design %>% 
  dplyr::rename(sample=sample_id)
samples_metadata(bestModel) <- design

# Add Feature Names
updated_features_names <- features_names(bestModel)
updated_features_names[["proteomics"]] <- protFeatures
updated_features_names[["phosphoproteomics"]] <- phosphoFeatures
updated_features_names[["transcriptomics"]] <- geneFeatures
updated_features_names[["immuneinfiltration"]] <- immuneFeat

features_names(bestModel) <- updated_features_names
```
```{r}
#| eval: !expr (!params$used_analysis)
slotNames(bestModel)
```

#### Save model
```{r}
#| eval: !expr (!params$used_analysis)
saveRDS(object = bestModel,file = '/multiomics/results/MOFA/LNposvsLNneg/Ductal/model.RDS',compress = 'gzip')
```

#### Correlation between factors
```{r}
#| eval: !expr (!params$used_analysis)
factor_corr_plot <- {plot_factor_cor(object = bestModel)
  grDevices::recordPlot()}

ggsave(filename = 'factor_corr_plot.png',plot = grDevices::replayPlot(factor_corr_plot),device = 'png',path = '/multiomics/results/figures/MOFA/LNposvsLNneg/Ductal/')
```

#### Variance decomposition
```{r}
#| eval: !expr (!params$used_analysis)
plotVariance <- plot_variance_explained(bestModel,max_r2 = 10)

ggsave(filename = 'MOFA_variance_allgroups.pdf',plot = plotVariance,device = 'pdf',path = '/multiomics/results/figures/MOFA/LNposvsLNneg/Ductal/',dpi = 600,width = 10,height = 10,units = 'in')
```

#### Factor correlation
```{r}
#| eval: !expr (!params$used_analysis)
covariatePlot <- correlate_factors_with_covariates(bestModel, 
  covariates = c('Group.Info','Size.mm','NCN.PAM50','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","PR",'SSP.PAM50',"NHG",'BCFi_event','tumor_mutational_burden','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma"), 
  plot="log_pval",
  abs = TRUE,
  alpha = 0.05
)

ggsave(filename = 'covariance.pdf',plot = covariatePlot,device = 'pdf',path = '/multiomics/results/figures/MOFA/LNposvsLNneg/Ductal/',dpi = 600,width = 10,height = 10,units = 'in')
```

#### Plot feature weights
```{r}
#| eval: !expr (!params$used_analysis)
immune_weights <- plot_top_weights(object = bestModel,
                                   view = 'immuneinfiltration',
                                   factors = 3,
                                   scale = TRUE,
                                   sign = 'all')

prot_weights <- plot_top_weights(bestModel,
                 view = "proteomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)

phospho_weigths <- plot_top_weights(bestModel,
                 view = "phosphoproteomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)
rna_weights <- plot_top_weights(bestModel,
                 view = "transcriptomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)

figPath <- c('prot_weights.png','phospho_weigths.png','rna_weights.png','immune_weights.png')
weightPlots <- list(prot_weights,phospho_weigths,rna_weights,immune_weights)
map2(.x = weightPlots,.y = figPath,.f = ~ggsave(filename = .y,plot = .x,device = 'png',path = '/multiomics/results/figures/MOFA/LNposvsLNneg/Ductal/',dpi = 300,width = 15,height = 15,units = 'in'))
```

#### Plot Heatmaps
```{r}
#| eval: !expr (!params$used_analysis)
prot_heatmap <- plot_data_heatmap(bestModel, 
  view = "proteomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

rna_heatmap <- plot_data_heatmap(bestModel, 
  view = "transcriptomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

phospho_heatmap <- plot_data_heatmap(bestModel, 
  view = "phosphoproteomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

immune_heatmap <- plot_data_heatmap(bestModel, 
  view = "immuneinfiltration",
  factor = 3,  
  features = 8,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)


figPath <- c('prot_heatmap.png','phospho_heatmap.png','rna_heatmap.png','immune_heatmap')
heatmapPlots <- list(prot_heatmap,phospho_heatmap,rna_heatmap,immune_heatmap)
map2(.x = heatmapPlots,.y = figPath,.f = ~ggsave(filename = .y,plot = .x,device = 'png',path = '/multiomics/results/figures/MOFA/LNposvsLNneg/Ductal/',dpi = 300,width = 15,height = 15,units = 'in'))
```

## Lymphnode Positive vs Negative - Lobular
```{r}
#| eval: !expr (!params$used_analysis)
varskeep <- c('workers','madFiltering','NAFiltering','params')
rm(list = c(setdiff(ls(),varskeep)))
invisible(gc())

if(!dir.exists(paths = '/multiomics/results/figures/MOFA/LNposvsLNneg/Lobular/')) {
  dir.create(path = '/multiomics/results/figures/MOFA/LNposvsLNneg/Lobular/',recursive = TRUE)
}

if(!dir.exists(paths = '/multiomics/results/MOFA/LNposvsLNneg/Lobular/')) {
  dir.create(path = '/multiomics/results/MOFA/LNposvsLNneg/Lobular/',recursive = TRUE)
}
```

### Data preprocessing and multiomics dataframe creation
```{r}
#| eval: !expr (!params$used_analysis)
design <- read_tsv('/multiomics/results/design_files/design_multiomics_mofa.tsv') %>% 
    filter(str_detect(string = InvCa.type,pattern = 'Lobular'))
proteindata <- read_tsv(file = '/multiomics/results/batchcorrection/fullproteome/cycloess_rollup_genes_batchcorrected.tsv')
prot <- proteindata %>% 
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Protein',cutoff = 0.75)
prot$Unique.Ids <- make.names(names = prot$Protein.Names,unique = TRUE)
protFeatures <- prot$Unique.Ids
prot <- prot %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
rownames(prot) <- protFeatures

phosphodata <- read_tsv(file = '/multiomics/results/batchcorrection/phosphoproteome/cycloess_peptide_flanking_batchcorrected.tsv')
phospho <- phosphodata %>%
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Modified.Sequence',cutoff = 0.5)
phospho$Unique.Ids <- make.names(names = phospho$Protein.Names,unique = TRUE)
phosphoFeatures <- phospho$Unique.Ids
phospho <- phospho %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
rownames(phospho) <- phosphoFeatures

genedata <- read_tsv(file = '/multiomics/data/transcriptomics/genematrix_adjusted_cycloess.tsv')
gene <- genedata %>% 
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Gene.ID',cutoff = 0.75)
gene$Unique.Ids <- make.names(names = gene$Gene.Name,unique = TRUE)
geneFeatures <- gene$Unique.Ids
gene <- gene %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix

immuneInfiltration <- read_tsv(file = '/multiomics/data/Immune_infiltration/combined_immuneinfiltration.tsv')
immuneFeat <- immuneInfiltration[['cell_type']]
immune <- immuneInfiltration %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix


dataMix <- list(proteomics=prot,
                phosphoproteomics=phospho,
                transcriptomics=gene,
                immuneinfiltration=immune)

response <- design %>% 
  dplyr::select(sample_id,LN)
```

### Model Parameters
#### Sizes

```{r}
#| eval: !expr (!params$used_analysis)
purrr::map(.x = dataMix,.f = dim)
```

#### Create MOFA object

```{r}
#| eval: !expr (!params$used_analysis)
MOFAobject <- create_mofa(dataMix)
```

#### Plot data overview

```{r}
#| eval: !expr (!params$used_analysis)
plot_data_overview(MOFAobject)
```

#### Data options

```{r}
#| eval: !expr (!params$used_analysis)
data_opts <- get_default_data_options(MOFAobject)
data_opts$scale_views <- TRUE
```

#### Model options

```{r}
#| eval: !expr (!params$used_analysis)
model_opts <- get_default_model_options(MOFAobject)
model_opts$num_factors <- 20
model_opts$spikeslab_weights <- TRUE
model_opts$ard_weights <- TRUE
```

#### Training options

```{r}
#| eval: !expr (!params$used_analysis)
train_opts <- get_default_training_options(MOFAobject)
train_opts$convergence_mode <- "slow"
```

### Model Training

```{r}
#| eval: !expr (!params$used_analysis)
set.seed(123)
MOFAobject <- prepare_mofa(MOFAobject,data_options = data_opts,model_options = model_opts,training_options = train_opts)

trainedModels <- future_map(
  .options = furrr_options(seed = TRUE,globals = TRUE),
  .x = 1:10,
  .f = function(x){
    MOFAobject@training_options$seed <- (123+x)
    run_mofa(MOFAobject,use_basilisk = TRUE)
    }
  )

bestModel <- select_model(trainedModels)
```

### Model evaluation
#### Add metadata

```{r}
#| eval: !expr (!params$used_analysis)
# Add metadata
design <- design %>% 
  dplyr::rename(sample=sample_id)
samples_metadata(bestModel) <- design

# Add Feature Names
updated_features_names <- features_names(bestModel)
updated_features_names[["proteomics"]] <- protFeatures
updated_features_names[["phosphoproteomics"]] <- phosphoFeatures
updated_features_names[["transcriptomics"]] <- geneFeatures
updated_features_names[["immuneinfiltration"]] <- immuneFeat

features_names(bestModel) <- updated_features_names
```
```{r}
#| eval: !expr (!params$used_analysis)
slotNames(bestModel)
```

#### Save model
```{r}
#| eval: !expr (!params$used_analysis)
saveRDS(object = bestModel,file = '/multiomics/results/MOFA/LNposvsLNneg/Lobular/model.RDS',compress = 'gzip')
```

#### Correlation between factors
```{r}
#| eval: !expr (!params$used_analysis)
factor_corr_plot <- {plot_factor_cor(object = bestModel)
  grDevices::recordPlot()}

ggsave(filename = 'factor_corr_plot.png',plot = grDevices::replayPlot(factor_corr_plot),device = 'png',path = '/multiomics/results/figures/MOFA/LNposvsLNneg/Lobular/')
```

#### Variance decomposition
```{r}
#| eval: !expr (!params$used_analysis)
plotVariance <- plot_variance_explained(bestModel,max_r2 = 10)

ggsave(filename = 'MOFA_variance_allgroups.pdf',plot = plotVariance,device = 'pdf',path = '/multiomics/results/figures/MOFA/LNposvsLNneg/Lobular/',dpi = 600,width = 10,height = 10,units = 'in')
```

#### Factor correlation
```{r}
#| eval: !expr (!params$used_analysis)
covariatePlot <- correlate_factors_with_covariates(bestModel, 
  covariates = c('Group.Info','Size.mm','NCN.PAM50','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","PR",'SSP.PAM50',"NHG",'BCFi_event','tumor_mutational_burden','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma"), 
  plot="log_pval",
  abs = TRUE,
  alpha = 0.05
)

ggsave(filename = 'covariance.pdf',plot = covariatePlot,device = 'pdf',path = '/multiomics/results/figures/MOFA/LNposvsLNneg/Lobular/',dpi = 600,width = 10,height = 10,units = 'in')
```

#### Plot feature weights
```{r}
#| eval: !expr (!params$used_analysis)
immune_weights <- plot_top_weights(object = bestModel,
                                   view = 'immuneinfiltration',
                                   factors = 3,
                                   scale = TRUE,
                                   sign = 'all')

prot_weights <- plot_top_weights(bestModel,
                 view = "proteomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)

phospho_weigths <- plot_top_weights(bestModel,
                 view = "phosphoproteomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)
rna_weights <- plot_top_weights(bestModel,
                 view = "transcriptomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)

figPath <- c('prot_weights.png','phospho_weigths.png','rna_weights.png','immune_weights.png')
weightPlots <- list(prot_weights,phospho_weigths,rna_weights,immune_weights)
map2(.x = weightPlots,.y = figPath,.f = ~ggsave(filename = .y,plot = .x,device = 'png',path = '/multiomics/results/figures/MOFA/LNposvsLNneg/Lobular/',dpi = 300,width = 15,height = 15,units = 'in'))
```

#### Plot Heatmaps
```{r}
#| eval: !expr (!params$used_analysis)
prot_heatmap <- plot_data_heatmap(bestModel, 
  view = "proteomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

rna_heatmap <- plot_data_heatmap(bestModel, 
  view = "transcriptomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

phospho_heatmap <- plot_data_heatmap(bestModel, 
  view = "phosphoproteomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

immune_heatmap <- plot_data_heatmap(bestModel, 
  view = "immuneinfiltration",
  factor = 3,  
  features = 8,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)


figPath <- c('prot_heatmap.png','phospho_heatmap.png','rna_heatmap.png','immune_heatmap')
heatmapPlots <- list(prot_heatmap,phospho_heatmap,rna_heatmap,immune_heatmap)
map2(.x = heatmapPlots,.y = figPath,.f = ~ggsave(filename = .y,plot = .x,device = 'png',path = '/multiomics/results/figures/MOFA/LNposvsLNneg/Lobular/',dpi = 300,width = 15,height = 15,units = 'in'))
```


## All samples - Group2 Excluded
```{r}
#| eval: !expr (params$used_analysis)
if(!dir.exists(paths = '/multiomics/results/figures/MOFA/AllSamples_noG2')) {
  dir.create(path = '/multiomics/results/figures/MOFA/AllSamples_noG2/',recursive = TRUE)
}

if(!dir.exists(paths = '/multiomics/results/MOFA/AllSamples_noG2/')) {
  dir.create(path = '/multiomics/results/MOFA/AllSamples_noG2/',recursive = TRUE)
}
```

### Data preprocessing and multiomics dataframe creation
```{r}
#| eval: !expr (params$used_analysis)
design <- read_tsv('/multiomics/results/design_files/design_multiomics_mofa.tsv') %>% 
  filter(str_detect(string = Group.Info,pattern = 'Group2',negate = TRUE))

proteindata <- read_tsv(file = '/multiomics/results/batchcorrection/fullproteome/cycloess_rollup_genes_batchcorrected.tsv')
prot <- proteindata %>% 
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Protein',cutoff = 0.75)
prot$Unique.Ids <- make.names(names = prot$Protein.Names,unique = TRUE)
protFeatures <- prot$Unique.Ids
protIds <- prot %>% 
  select(matches("Name|Gene|Ids|Sequence|Phosphosite|Residue"))
prot <- prot %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
rownames(prot) <- protFeatures

phosphodata <- read_tsv(file = '/multiomics/results/batchcorrection/phosphoproteome/cycloess_peptide_flanking_batchcorrected.tsv')
phospho <- phosphodata %>%
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Modified.Sequence',cutoff = 0.5)
phospho$Unique.Ids <- make.names(names = phospho$Protein.Names,unique = TRUE)
phosphoFeatures <- phospho$Unique.Ids
phosphoIds <- phospho %>% 
  select(matches("Name|Gene|Ids|Sequence|Phosphosite|Residue"))
phospho <- phospho %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix
rownames(phospho) <- phosphoFeatures


genedata <- read_tsv(file = '/multiomics/data/transcriptomics/genematrix_adjusted_cycloess.tsv')
gene <- genedata %>% 
  NAFiltering(design = design,sampleCol = 'sample_id',cutoff = 0.3) %>% 
  madFiltering(design = design,sampleCol = 'sample_id',annotation = 'Gene.ID',cutoff = 0.75)
gene$Unique.Ids <- make.names(names = gene$Gene.Name,unique = TRUE)
geneFeatures <- gene$Unique.Ids
geneIds <- gene %>% 
  select(matches("Name|Gene|Ids|Sequence"))
gene <- gene %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix

immuneInfiltration <- read_tsv(file = '/multiomics/data/Immune_infiltration/combined_immuneinfiltration.tsv')
immuneFeat <- immuneInfiltration[['cell_type']]
immune <- immuneInfiltration %>% 
  dplyr::select(design$sample_id) %>% 
  as.matrix


dataMix <- list(proteomics=prot,
                phosphoproteomics=phospho,
                transcriptomics=gene,
                immuneinfiltration=immune)

response <- design %>% 
  dplyr::select(sample_id,LN)

allIds <- list(proteomics=protIds,phosphoproteomics=phosphoIds,transcriptomics=geneIds) %>% 
  map2(.y = c('proteomics','phosphoproteomics','transcriptomics'),.f = function(x,y){
    data <- x %>% mutate(across(Unique.Ids,~paste0(.x,'.',y)))
  }) %>% 
  list_rbind()

write_tsv(x = allIds,file = '/multiomics/results/MOFA/AllSamples_noG2/featIDs.tsv')
```

### Model Parameters
#### Sizes

```{r}
#| eval: !expr (params$used_analysis)
purrr::map(.x = dataMix,.f = dim)
```

#### Create MOFA object

```{r}
#| eval: !expr (params$used_analysis)
MOFAobject <- create_mofa(dataMix)
```

#### Plot data overview

```{r}
#| eval: !expr (params$used_analysis)
plot_data_overview(MOFAobject)
```

#### Data options

```{r}
#| eval: !expr (params$used_analysis)
data_opts <- get_default_data_options(MOFAobject)
data_opts$scale_views <- TRUE
```

#### Model options

```{r}
#| eval: !expr (params$used_analysis)
model_opts <- get_default_model_options(MOFAobject)
model_opts$num_factors <- 20
model_opts$spikeslab_weights <- TRUE
model_opts$ard_weights <- TRUE
```

#### Training options

```{r}
#| eval: !expr (params$used_analysis)
train_opts <- get_default_training_options(MOFAobject)
train_opts$convergence_mode <- "slow"
```

### Model Training

```{r}
#| eval: !expr (params$used_analysis)
set.seed(123)
MOFAobject <- prepare_mofa(MOFAobject,data_options = data_opts,model_options = model_opts,training_options = train_opts)

trainedModels <- future_map(
  .options = furrr_options(seed = TRUE,globals = TRUE),
  .x = 1:10,
  .f = function(x){
    MOFAobject@training_options$seed <- (123+x)
    run_mofa(MOFAobject,use_basilisk = TRUE)
    }
  )

bestModel <- select_model(trainedModels)
```

### Model evaluation
#### Add metadata

```{r}
#| eval: !expr (params$used_analysis)
# Add metadata
design <- design %>% 
  dplyr::rename(sample=sample_id)
samples_metadata(bestModel) <- design

# Add Feature Names
updated_features_names <- features_names(bestModel)
updated_features_names[["proteomics"]] <- protFeatures
updated_features_names[["phosphoproteomics"]] <- phosphoFeatures
updated_features_names[["transcriptomics"]] <- geneFeatures
updated_features_names[["immuneinfiltration"]] <- immuneFeat

features_names(bestModel) <- updated_features_names
```
```{r}
#| eval: !expr (params$used_analysis)
slotNames(bestModel)
```

#### Save model
```{r}
#| eval: !expr (params$used_analysis)
saveRDS(object = bestModel,file = '/multiomics/results/MOFA/AllSamples_noG2/model.RDS',compress = 'gzip')
```

#### Correlation between factors
```{r}
#| eval: !expr (params$used_analysis)
factor_corr_plot <- {plot_factor_cor(object = bestModel)
  grDevices::recordPlot()}

ggsave(filename = 'factor_corr_plot.png',plot = grDevices::replayPlot(factor_corr_plot),device = 'png',path = '/multiomics/results/figures/MOFA/AllSamples_noG2/')
```

#### Variance decomposition
```{r}
#| eval: !expr (params$used_analysis)
plotVariance <- plot_variance_explained(bestModel,max_r2 = 10)

ggsave(filename = 'MOFA_variance_allgroups.pdf',plot = plotVariance,device = 'pdf',path = '/multiomics/results/figures/MOFA/AllSamples_noG2/',dpi = 600,width = 10,height = 10,units = 'in')
```

#### Factor correlation
```{r}
#| eval: !expr (params$used_analysis)
covariatePlot <- correlate_factors_with_covariates(bestModel, 
  covariates = c('Group.Info','Size.mm','NCN.PAM50','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","PR",'SSP.PAM50',"NHG",'BCFi_event','tumor_mutational_burden','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma"), 
  plot="log_pval",
  abs = TRUE,
  alpha = 0.05
)

ggsave(filename = 'covariance.pdf',plot = covariatePlot,device = 'pdf',path = '/multiomics/results/figures/MOFA/AllSamples_noG2/',dpi = 600,width = 10,height = 10,units = 'in')
```

#### Plot feature weights
```{r}
#| eval: !expr (params$used_analysis)
immune_weights <- plot_top_weights(object = bestModel,
                                   view = 'immuneinfiltration',
                                   factors = 3,
                                   scale = TRUE,
                                   sign = 'all')

prot_weights <- plot_top_weights(bestModel,
                 view = "proteomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)

phospho_weigths <- plot_top_weights(bestModel,
                 view = "phosphoproteomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)
rna_weights <- plot_top_weights(bestModel,
                 view = "transcriptomics",
                 factor = 3,
                 nfeatures = 100,     # Top number of features to highlight
                 scale = T           # Scale weights from -1 to 1
)

figPath <- c('prot_weights.png','phospho_weigths.png','rna_weights.png','immune_weights.png')
weightPlots <- list(prot_weights,phospho_weigths,rna_weights,immune_weights)
map2(.x = weightPlots,.y = figPath,.f = ~ggsave(filename = .y,plot = .x,device = 'png',path = '/multiomics/results/figures/MOFA/AllSamples_noG2/',dpi = 300,width = 15,height = 15,units = 'in'))
```

#### Plot Heatmaps
```{r}
#| eval: !expr (params$used_analysis)
prot_heatmap <- plot_data_heatmap(bestModel, 
  view = "proteomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

rna_heatmap <- plot_data_heatmap(bestModel, 
  view = "transcriptomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

phospho_heatmap <- plot_data_heatmap(bestModel, 
  view = "phosphoproteomics",
  factor = 3,  
  features = 100,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)

immune_heatmap <- plot_data_heatmap(bestModel, 
  view = "immuneinfiltration",
  factor = 3,  
  features = 8,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = FALSE,
  scale = "row",
  denoise = TRUE,
  annotation_samples = c('Group.Info','DRFi_event',"TreatGroup","Age.(5-year.range,.e.g.,.35(31-35),.40(36-40),.45(41-45).etc.)","NCN.ROR.risk.cat","NHG",'BCFi_event','tumor_mutational_burden','InvCa.type','LN','LN.spec',"HisScoreFat","HisScoreInsituCancer","HisScoreInvasiveCancer","HisScoreLymphocytes","HisScoreNormal","HisScoreStroma")
)


figPath <- c('prot_heatmap.png','phospho_heatmap.png','rna_heatmap.png','immune_heatmap')
heatmapPlots <- list(prot_heatmap,phospho_heatmap,rna_heatmap,immune_heatmap)
map2(.x = heatmapPlots,.y = figPath,.f = ~ggsave(filename = .y,plot = .x,device = 'png',path = '/multiomics/results/figures/MOFA/AllSamples_noG2/',dpi = 300,width = 15,height = 15,units = 'in'))
```

```{r}
#| eval: !expr (params$used_analysis)
rm(list = ls())
invisible(gc())
```
