---
title: "Table 1 - Data Completeness"
author: "Sergio Mosquim Junior"
format:
  html:
    toc: true
    code-fold: true
    embed-resources: true
    toc-location: left
    smooth-scroll: true
engine: knitr
---

# Requires Packages:
```{r required packages}
#| message: false
#| warning: false
library(tidyverse)
```

# Loading Necessary Data
## Proteomics
```{r}
prot <- read_tsv(file = "/home/multiomics/results/rollup/fullproteome/cycloess_Rollup.tsv")
design_prot <- read_tsv(file = "/home/multiomics/results/design_files/design_Full_noPool.tsv")
```

## Phosphoproteomics
```{r}
design_phospho <- read_tsv("/home/multiomics/results/design_files/design_phospho_noPool.tsv")
phospho <- read_tsv("/home/multiomics/results/normalization/phosphoproteome/peptide_normalization/CycLoess-normalized.txt")
```

## Transcriptomics
```{r}
design_gene <- read_tsv("/home/multiomics/results/design_files/design_RNA_noPool.tsv")
gene <- read_tsv("/home/multiomics/results/normalization/transcriptomics/genematrix_adjusted_cycloess.tsv")
```

# Table Generation
```{r}
prot_process <- prot %>% rowwise() %>% mutate(na_percent=sum(is.na(c_across(starts_with("S"))))/177)
prot_filter <- prot_process %>% filter(na_percent<=0.3)

phospho_process <- phospho %>% rowwise() %>% mutate(na_percent=sum(is.na(c_across(starts_with("S0"))))/162)
phospho_filter <- phospho_process %>% filter(na_percent<=0.3)

gene_process <- gene %>% rowwise() %>% mutate(na_percent=sum(is.na(c_across(starts_with("S0"))))/182)
gene_filter <- gene_process %>% filter(na_percent<=0.3)

table_data <- data.frame(OMICS=c("Proteomics DIA (protein groups)", 
                             "Phosphoproteomics DIA (peptides)", 
                             "Transcriptomics (genes)"),
                     `Number of Features`=c(nrow(prot_process),
                                            nrow(phospho_process),
                                            nrow(gene_process)),
                     `At Least 70% D.C.`=c(nrow(prot_filter),
                                           nrow(phospho_filter),
                                           nrow(gene_filter)),
                     `Complete Observations`=c(nrow(drop_na(prot_filter)),
                                               nrow(drop_na(phospho_filter)),
                                               nrow(drop_na(gene_filter))))

knitr::kable(x = table_data,format = "html",caption = "Table 1: Overview of the number of features identified in all three omics layers",label = "D.C.: Data Completeness")
```

